---
title: "1. Introduction"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{1. Introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)

options(crayon.enabled = F)
```

```{r setup, warning=FALSE}
library(tickTack)
require(dplyr)
```

---
title: "Introduction"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

The `tickTack` package provides tools for timing the occurrence of copy number alterations (CNA). This vignette introduces the structure of the input data required for `tickTack` functions, using the example dataset `tickTack::pcawg_example`.

## Input Data Structure

The input data consists of three components stored as named elements in a list: `mutations`, `cna`, and `metadata`. Below is a description of each component.

### 1. Mutations

The `mutations` component is a tibble containing information about somatic mutations. Each row represents a mutation, with the following columns:

- **`chr`**: Chromosome where the mutation occurs.
- **`from`** and **`to`**: Start and end positions of the mutation on the chromosome.
- **`ref`** and **`alt`**: Reference and alternate alleles.
- **`DP`**: Depth of sequencing coverage at the mutation site.
- **`NV`**: Number of reads supporting the variant.
- **`VAF`**: Variant allele frequency, calculated as `NV / DP`.
- **`sample`**: Unique identifier for the sample.

For example, the first few rows of `mutations` look like this:

```{r echo=TRUE}
tickTack::pcawg_example$mutations %>% head()
```

### 2. Copy Number Alterations (CNA)

The `cna` component is a tibble that describes regions of the genome with alterations in copy number. Each row represents a genomic segment, with the following columns:

- **`chr`**: Chromosome of the segment.
- **`from`** and **`to`**: Start and end positions of the segment.
- **`Major`** and **`minor`**: Major and minor allele copy numbers.
- **`CCF`** Cancer cell fraction for the segment.
- **`total_cn`**: Total copy number (sum of **`Major`** and **`minor`**).

 Here is the preview of the `cna` data:

```{r echo=TRUE}
tickTack::pcawg_example$cna %>% head()
```

### 3. Metadata

The `metadata` component is a tibble containing sample-level information, with the following columns:

- **`sample`**: Unique identifier for the sample.
- **`purity`**: Tumor purity, representing the proportion of cancer cells in the sample.
- **`ploidy`**: Average ploidy of the sample.
- **`purity_conf_mad`**: Confidence interval for the purity estimate.
- **`wgd_status`**: Whole genome doubling status (e.g., **`wgd`** or **`no wgd`**).
- **`wgd_uncertain`**: Logical indicating uncertainty in the **`wgd_status`**.

An example of the `metadata is shown below`:

```{r echo=TRUE}
tickTack::pcawg_example$metadata
```

## Types of Copy Number Alterations (CNAs)

The `cna` component can include different types of copy number segments, categorized as follows:

### Clonal Simple CNAs
These are straightforward alterations with specific copy number configurations:

- **`1:0`**: Loss of heterozygosity (LOH).
- **`2:0`**: Copy neutral LOH.
- **`1:1`**: Diploid heterozygous (assumed to be the normal reference).
- **`2:1`**: Trisomy.
- **`2:2`**: Tetraploidy.

### Clonal Complex and Subclonal CNAs
Clonal complex CNAs include any clonal CNA that is not considered simple.
On the other hand, subclonal CNAs involve a mixture of two subclones, where each subclone is defined by a simple CNA.

### Timing Eligibility
Only **clonal CNAs** with specific configurations can be timed:

- **`2:1`**: Trisomy.
- **`2:0`**: Copy neutral LOH.
- **`2:2`**: Tetraploidy.

> Note : Only **clonal CNAs** are accepted by `tickTack`. Subclonal CNAs or other complex configurations are not supported for timing analysis.

## Reference genome coordinates

tickTack uses a genome coordinates reference system to convert relative relative to absolute coordinates, a step required to plot segments across the whole genome. For instance, if a mutation maps to position $100$ of chromosome `chr2`, its absolute coordinate is $100 + L$ where $L$ is the length of `chr1`. The reference system adopted by tickTack needs therefore to report the length of each chromosome, plus the information regarding the boundary of each centromere.

> Note: mapping of mutations onto segments is independent of the reference genome, and it will work as far as both mutation and CNA segments are mapped to the same reference.

tickTack supports two coordinates reference genomes:

* `hg19` or `GRCh37`;
* `hg38` or `GRCh38` (default),

for which two dataframes are stored inside the package.

```{r, fig.width=9.5, fig.height=3}
tickTack::chr_coordinates_hg19

tickTack::chr_coordinates_GRCh38
```
