<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Introduction</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(tickTack)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">require</span>(dplyr)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co">#&gt; Loading required package: dplyr</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co">#&gt; Attaching package: &#39;dplyr&#39;</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:stats&#39;:</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="co">#&gt;     filter, lag</span></span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="co">#&gt; The following objects are masked from &#39;package:base&#39;:</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<p>The <code>tickTack</code> package provides tools for timing the
occurrence of copy number alterations (CNA). This vignette introduces
the structure of the input data required for <code>tickTack</code>
functions, using the example dataset
<code>tickTack::pcawg_example</code>.</p>
<div id="input-data-structure" class="section level2">
<h2>Input Data Structure</h2>
<p>The input data consists of three components stored as named elements
in a list: <code>mutations</code>, <code>cna</code>, and
<code>metadata</code>. Below is a description of each component.</p>
<div id="mutations" class="section level3">
<h3>1. Mutations</h3>
<p>The <code>mutations</code> component is a tibble containing
information about somatic mutations. Each row represents a mutation,
with the following columns:</p>
<ul>
<li><strong><code>chr</code></strong>: Chromosome where the mutation
occurs.</li>
<li><strong><code>from</code></strong> and
<strong><code>to</code></strong>: Start and end positions of the
mutation on the chromosome.</li>
<li><strong><code>ref</code></strong> and
<strong><code>alt</code></strong>: Reference and alternate alleles.</li>
<li><strong><code>DP</code></strong>: Depth of sequencing coverage at
the mutation site.</li>
<li><strong><code>NV</code></strong>: Number of reads supporting the
variant.</li>
<li><strong><code>VAF</code></strong>: Variant allele frequency,
calculated as <code>NV / DP</code>.</li>
<li><strong><code>sample</code></strong>: Unique identifier for the
sample.</li>
</ul>
<p>For example, the first few rows of <code>mutations</code> look like
this:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>tickTack<span class="sc">::</span>pcawg_example<span class="sc">$</span>mutations <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 9</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt;   chr      from      to ref   alt      DP    NV   VAF sample                    </span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;   &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;                     </span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co">#&gt; 1 chr1  1015594 1015594 C     C        99    16 0.162 00db1b95-8ca3-4cc4-bb46-6…</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; 2 chr1  1866371 1866371 C     C       250    67 0.268 00db1b95-8ca3-4cc4-bb46-6…</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; 3 chr1  1921712 1921712 C     C        62    20 0.323 00db1b95-8ca3-4cc4-bb46-6…</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; 4 chr1  2049858 2049858 G     G       118    15 0.127 00db1b95-8ca3-4cc4-bb46-6…</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; 5 chr1  2357842 2357842 C     C        84    12 0.143 00db1b95-8ca3-4cc4-bb46-6…</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt; 6 chr1  2771915 2771915 G     G        90    27 0.3   00db1b95-8ca3-4cc4-bb46-6…</span></span></code></pre></div>
</div>
<div id="copy-number-alterations-cna" class="section level3">
<h3>2. Copy Number Alterations (CNA)</h3>
<p>The <code>cna</code> component is a tibble that describes regions of
the genome with alterations in copy number. Each row represents a
genomic segment, with the following columns:</p>
<ul>
<li><strong><code>chr</code></strong>: Chromosome of the segment.</li>
<li><strong><code>from</code></strong> and
<strong><code>to</code></strong>: Start and end positions of the
segment.</li>
<li><strong><code>Major</code></strong> and
<strong><code>minor</code></strong>: Major and minor allele copy
numbers.</li>
<li><strong><code>CCF</code></strong> Cancer cell fraction for the
segment.</li>
<li><strong><code>total_cn</code></strong>: Total copy number (sum of
<strong><code>Major</code></strong> and
<strong><code>minor</code></strong>).</li>
</ul>
<p>Here is the preview of the <code>cna</code> data:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>tickTack<span class="sc">::</span>pcawg_example<span class="sc">$</span>cna <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 7</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co">#&gt;   chr       from        to Major minor   CCF total_cn</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;    &lt;dbl&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;</span></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co">#&gt; 1 chr1     10001    790008     2     2 1            4</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="co">#&gt; 2 chr1    790009  13212499     2     2 1            4</span></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt; 3 chr1  13212500  33458785     2     2 1            4</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt; 4 chr1  33458786  33564126     2     2 0.194        4</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; 5 chr1  33564127  56834601     2     2 1            4</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; 6 chr1  56834602 121499999     2     2 1            4</span></span></code></pre></div>
</div>
<div id="metadata" class="section level3">
<h3>3. Metadata</h3>
<p>The <code>metadata</code> component is a tibble containing
sample-level information, with the following columns:</p>
<ul>
<li><strong><code>sample</code></strong>: Unique identifier for the
sample.</li>
<li><strong><code>purity</code></strong>: Tumor purity, representing the
proportion of cancer cells in the sample.</li>
<li><strong><code>ploidy</code></strong>: Average ploidy of the
sample.</li>
<li><strong><code>purity_conf_mad</code></strong>: Confidence interval
for the purity estimate.</li>
<li><strong><code>wgd_status</code></strong>: Whole genome doubling
status (e.g., <strong><code>wgd</code></strong> or
<strong><code>no wgd</code></strong>).</li>
<li><strong><code>wgd_uncertain</code></strong>: Logical indicating
uncertainty in the <strong><code>wgd_status</code></strong>.</li>
</ul>
<p>An example of the <code>metadata is shown below</code>:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>tickTack<span class="sc">::</span>pcawg_example<span class="sc">$</span>metadata</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 1 × 6</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt;   sample                  purity ploidy purity_conf_mad wgd_status wgd_uncertain</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                    &lt;dbl&gt;  &lt;dbl&gt;           &lt;dbl&gt; &lt;chr&gt;      &lt;lgl&gt;        </span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt; 1 00db1b95-8ca3-4cc4-bb4…  0.406   4.08           0.009 wgd        FALSE</span></span></code></pre></div>
</div>
</div>
<div id="types-of-copy-number-alterations-cnas" class="section level2">
<h2>Types of Copy Number Alterations (CNAs)</h2>
<p>The <code>cna</code> component can include different types of copy
number segments, categorized as follows:</p>
<div id="clonal-simple-cnas" class="section level3">
<h3>Clonal Simple CNAs</h3>
<p>These are straightforward alterations with specific copy number
configurations:</p>
<ul>
<li><strong><code>1:0</code></strong>: Loss of heterozygosity
(LOH).</li>
<li><strong><code>2:0</code></strong>: Copy neutral LOH.</li>
<li><strong><code>1:1</code></strong>: Diploid heterozygous (assumed to
be the normal reference).</li>
<li><strong><code>2:1</code></strong>: Trisomy.</li>
<li><strong><code>2:2</code></strong>: Tetraploidy.</li>
</ul>
</div>
<div id="clonal-complex-and-subclonal-cnas" class="section level3">
<h3>Clonal Complex and Subclonal CNAs</h3>
<p>Clonal complex CNAs include any clonal CNA that is not considered
simple. On the other hand, subclonal CNAs involve a mixture of two
subclones, where each subclone is defined by a simple CNA.</p>
</div>
<div id="timing-eligibility" class="section level3">
<h3>Timing Eligibility</h3>
<p>Only <strong>clonal CNAs</strong> with specific configurations can be
timed:</p>
<ul>
<li><strong><code>2:1</code></strong>: Trisomy.</li>
<li><strong><code>2:0</code></strong>: Copy neutral LOH.</li>
<li><strong><code>2:2</code></strong>: Tetraploidy.</li>
</ul>
<blockquote>
<p>Note : Only <strong>clonal CNAs</strong> are accepted by
<code>tickTack</code>. Subclonal CNAs or other complex configurations
are not supported for timing analysis.</p>
</blockquote>
</div>
</div>
<div id="reference-genome-coordinates" class="section level2">
<h2>Reference genome coordinates</h2>
<p>tickTack uses a genome coordinates reference system to convert
relative relative to absolute coordinates, a step required to plot
segments across the whole genome. For instance, if a mutation maps to
position <span class="math inline">\(100\)</span> of chromosome
<code>chr2</code>, its absolute coordinate is <span class="math inline">\(100 + L\)</span> where <span class="math inline">\(L\)</span> is the length of <code>chr1</code>. The
reference system adopted by tickTack needs therefore to report the
length of each chromosome, plus the information regarding the boundary
of each centromere.</p>
<blockquote>
<p>Note: mapping of mutations onto segments is independent of the
reference genome, and it will work as far as both mutation and CNA
segments are mapped to the same reference.</p>
</blockquote>
<p>tickTack supports two coordinates reference genomes:</p>
<ul>
<li><code>hg19</code> or <code>GRCh37</code>;</li>
<li><code>hg38</code> or <code>GRCh38</code> (default),</li>
</ul>
<p>for which two dataframes are stored inside the package.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>tickTack<span class="sc">::</span>chr_coordinates_hg19</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 24 × 6</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#&gt;    chr      length       from         to centromerStart centromerEnd</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;     &lt;int&gt;      &lt;dbl&gt;      &lt;dbl&gt;          &lt;dbl&gt;        &lt;dbl&gt;</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt;  1 chr1  249250621          0  249250621      121535434    124535434</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt;  2 chr2  243199373  249250621  492449994      341576792    344576792</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt;  3 chr3  198022430  492449994  690472424      582954848    585954848</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt;  4 chr4  191154276  690472424  881626700      740132541    743132541</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt;  5 chr5  180915260  881626700 1062541960      928032341    931032341</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt;  6 chr6  171115067 1062541960 1233657027     1121372126   1124372126</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt;  7 chr7  159138663 1233657027 1392795690     1291711358   1294711358</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co">#&gt;  8 chr8  146364022 1392795690 1539159712     1436634577   1439634577</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="co">#&gt;  9 chr9  141213431 1539159712 1680373143     1586527391   1589527391</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co">#&gt; 10 chr10 135534747 1680373143 1815907890     1719628078   1722628078</span></span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="co">#&gt; # ℹ 14 more rows</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>tickTack<span class="sc">::</span>chr_coordinates_GRCh38</span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a><span class="co">#&gt; # A tibble: 24 × 6</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a><span class="co">#&gt;    chr      length       from         to centromerStart centromerEnd</span></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;     &lt;dbl&gt;      &lt;dbl&gt;      &lt;dbl&gt;          &lt;dbl&gt;        &lt;dbl&gt;</span></span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a><span class="co">#&gt;  1 chr1  248956422          0  248956422      122026459    124849229</span></span>
<span id="cb5-22"><a href="#cb5-22" tabindex="-1"></a><span class="co">#&gt;  2 chr2  242193529  248956422  491149951      341144567    341144567</span></span>
<span id="cb5-23"><a href="#cb5-23" tabindex="-1"></a><span class="co">#&gt;  3 chr3  198295559  491149951  689445510      581922409    582703370</span></span>
<span id="cb5-24"><a href="#cb5-24" tabindex="-1"></a><span class="co">#&gt;  4 chr4  190214555  689445510  879660065      739157571    739157571</span></span>
<span id="cb5-25"><a href="#cb5-25" tabindex="-1"></a><span class="co">#&gt;  5 chr5  181538259  879660065 1061198324      926145965    929381368</span></span>
<span id="cb5-26"><a href="#cb5-26" tabindex="-1"></a><span class="co">#&gt;  6 chr6  170805979 1061198324 1232004303     1119752212   1119752212</span></span>
<span id="cb5-27"><a href="#cb5-27" tabindex="-1"></a><span class="co">#&gt;  7 chr7  159345973 1232004303 1391350276     1290173956   1293382091</span></span>
<span id="cb5-28"><a href="#cb5-28" tabindex="-1"></a><span class="co">#&gt;  8 chr8  145138636 1391350276 1536488912     1435384020   1435384020</span></span>
<span id="cb5-29"><a href="#cb5-29" tabindex="-1"></a><span class="co">#&gt;  9 chr9  138394717 1536488912 1674883629     1579878547   1579878547</span></span>
<span id="cb5-30"><a href="#cb5-30" tabindex="-1"></a><span class="co">#&gt; 10 chr10 133797422 1674883629 1808681051     1714570311   1716429449</span></span>
<span id="cb5-31"><a href="#cb5-31" tabindex="-1"></a><span class="co">#&gt; # ℹ 14 more rows</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
