<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>3. Understanding Timing Clonal Peaks</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">3. Understanding Timing Clonal Peaks</h1>



<p>This vignette delves into how the <code>tickTack</code> package
obtains timing results for clonal peaks using the <code>fit</code>
function. It explains the underlying processes, including the
computation of clonal peaks and the Stan model used for inference.</p>
<div id="key-components-of-the-analysis" class="section level2">
<h2>Key Components of the Analysis</h2>
<p>The timing results are obtained in the following steps:</p>
<ol style="list-style-type: decimal">
<li><strong>Extracting clonal peaks</strong>: Theoretical peaks in the
Variant Allele Frequency (VAF) spectrum are calculated for specific
karyotypes and tumor purity.</li>
<li><strong>Filtering mutations</strong>: Mutations within confidence
intervals around these peaks are selected for analysis.</li>
<li><strong>Fitting the model</strong>: Bayesian inference is performed
using a Stan model to estimate the timing of clonal expansions.</li>
</ol>
<div id="step-1-calculating-clonal-peaks" class="section level3">
<h3>Step 1: Calculating Clonal Peaks</h3>
<p>The clonal peaks are calculated using the
<code>get_clonal_peaks</code> function. The function computes the
theoretical VAF peaks based on the karyotype and tumor purity. They are
computed using the following equation:</p>
<p><span class="math display">\[
\text{Peak}_i = \frac{M_i \cdot P}{(N_{\text{tot}} \cdot P) + 2 \cdot (1
- P)}
\]</span></p>
<p>Where: - <span class="math inline">\(M_i\)</span>: The multiplicity
of the allele of interest (e.g., the major allele or a single copy for
heterozygosity). - <span class="math inline">\(P\)</span>: The tumor
purity (a value between 0 and 1). - <span class="math inline">\(N_{\text{tot}}\)</span>: The total number of
copies in the karyotype (e.g., sum of the major and minor alleles). -
<span class="math inline">\(2\)</span>: Accounts for the two copies in
normal diploid cells.</p>
<div id="explanation" class="section level4">
<h4>Explanation</h4>
<ul>
<li>The <strong>numerator</strong>, <span class="math inline">\(M_i
\cdot P\)</span>, represents the contribution of the tumor cellâ€™s allele
multiplicity adjusted for tumor purity.</li>
<li>The <strong>denominator</strong>, <span class="math inline">\((N_{\text{tot}} \cdot P) + 2 \cdot (1 -
P)\)</span>, normalizes the peak to account for contributions from both
tumor and normal diploid cells in the sample.</li>
</ul>
</div>
<div id="example-calculation" class="section level4">
<h4>Example Calculation</h4>
<p>For a karyotype <span class="math inline">\(k =
&quot;2:1&quot;\)</span> (major allele = 2, minor allele = 1) and tumor
purity <span class="math inline">\(P = 0.4\)</span>:</p>
<ol style="list-style-type: decimal">
<li><span class="math inline">\(N_{\text{tot}} = 2 + 1 =
3\)</span>.</li>
<li>Compute peaks for <span class="math inline">\(M_i = 1\)</span>
(single allele) and <span class="math inline">\(M_i = 2\)</span> (major
allele): <span class="math display">\[
\text{Peak}_1 = \frac{1 \cdot 0.4}{(3 \cdot 0.4) + 2 \cdot (1 - 0.4)} =
\frac{0.4}{1.2 + 1.2} = 0.1667
\]</span> <span class="math display">\[
\text{Peak}_2 = \frac{2 \cdot 0.4}{(3 \cdot 0.4) + 2 \cdot (1 - 0.4)} =
\frac{0.8}{1.2 + 1.2} = 0.3333
\]</span></li>
</ol>
</div>
</div>
<div id="step-2-filtering-mutations" class="section level3">
<h3>Step 2: Filtering Mutations</h3>
<p>Once the peaks are obtained, mutations within a genomic segment are
filtered based on whether their VAF values fall within a confidence
interval around these peaks.</p>
<p>The confidence intervals are computed using either:</p>
<ul>
<li><strong>Binomial distribution</strong>: for standard VAF
estimation</li>
<li><strong>Beta-binomial distribution</strong>: to account for
overdispersion</li>
</ul>
<div id="filtering-logic" class="section level4">
<h4>Filtering logic</h4>
<p>For each mutation, the number of variant reads (NV) is compared to
the confidence interval calculated for the depth (DP) at each peak:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Example pseudo-code for filtering</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>probs <span class="ot">&lt;-</span> <span class="fu">c</span>(alpha <span class="sc">/</span> <span class="dv">2</span>, <span class="dv">1</span> <span class="sc">-</span> alpha <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="cf">for</span> (p <span class="cf">in</span> peaks) {</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>  quantiles <span class="ot">&lt;-</span> <span class="fu">qbinom</span>(probs, DP, p) <span class="co"># or qbb for beta-binomial</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  <span class="cf">if</span> (NV <span class="sc">&gt;=</span> quantiles[<span class="dv">1</span>] <span class="sc">&amp;&amp;</span> NV <span class="sc">&lt;=</span> quantiles[<span class="dv">2</span>]) {</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>    <span class="co"># Mutation is accepted</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  }</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>}</span></code></pre></div>
<p>Only mutations passing this test are used for model fitting.</p>
</div>
</div>
<div id="step-3-bayesian-inference-with-stan" class="section level3">
<h3>Step 3: Bayesian Inference with Stan</h3>
<p>The final step is to fit a Stan model to the filtered mutations. The
model estimates the posterior distribution of <code>tau</code>, which
represents the timing of clonal expansions.</p>
<p>The stan code used for the model is divided in the following
pieces.</p>
<div id="data-block" class="section level4">
<h4>Data block</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>data {</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  int N;</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  array[N] int NV;</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  array[N] int DP;</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  array[<span class="dv">2</span>] real peaks;</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>}</span></code></pre></div>
<p>Here, the data block defines inputs that the model uses:</p>
<ul>
<li><strong>N</strong>: Total number of mutations to analyze.</li>
<li><strong>NV</strong>: Number of variant reads observed for each
mutation.</li>
<li><strong>DP</strong>: Total sequencing depth for each mutation.</li>
<li><strong>peaks</strong>: An array of two real values representing the
theoretical clonal peaks (e.g., 2:1 and 2:2 ratios or other hypothesized
clonal frequencies).</li>
</ul>
</div>
<div id="parameters-block" class="section level4">
<h4>Parameters block</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>parameters {</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  simplex[<span class="dv">2</span>] omega;</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>}</span></code></pre></div>
<p>The parameters block contains:</p>
<ul>
<li><strong>omega</strong>: A vector representing the probability
contributions of each theoretical peak. It follows a Dirichlet
distribution (a standard approach for modeling probabilities that sum to
1).</li>
</ul>
</div>
<div id="model-block" class="section level4">
<h4>Model block</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>model {</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  vector[<span class="dv">2</span>] contributions;</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  omega <span class="sc">~</span> <span class="fu">dirichlet</span>(<span class="fu">rep_vector</span>(<span class="fl">2.0</span>, <span class="dv">2</span>));</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  </span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N) {</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>    <span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) {</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>      contributions[k] <span class="ot">=</span> <span class="fu">log</span>(omega[k]) <span class="sc">+</span> <span class="fu">binomial_lpmf</span>(NV[i] <span class="sc">|</span> DP[i], peaks[k]);</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>    }</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>    target <span class="sc">+</span><span class="er">=</span> <span class="fu">log_sum_exp</span>(contributions);</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  }</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>}</span></code></pre></div>
<p>The <code>model</code> block contains three components:</p>
<ul>
<li><strong>Prior</strong>: The line
<code>omega ~ dirichlet(rep_vector(2.0, 2));</code> applies a uniform
Dirichlet prior to the karyotype mixture probabilities (omega). This
means that we do not a priori favor one clonal peak over the other
unless there is data evidence.</li>
<li><strong>Likelihood</strong>: The likelihood is computed using the
<code>binomial_lpmf(NV[i] | DP[i], peaks[k])</code> (or the Negative
binomial). This term computes the probability of observing
<code>NV[i]</code> variant reads given sequencing depth
<code>DP[i]</code> and a given theoretical clonal peak value
<code>peaks[k]</code>.</li>
<li><strong>Marginalization</strong>: The line
<code>log_sum_exp(contributions)</code> computes the log marginal
likelihood by summing over possible contributions (karyotypes) using the
softmax approach, allowing the model to integrate over possible
karyotype contributions (<code>omega</code>).</li>
</ul>
</div>
</div>
<div id="final-step-conversion-from-omega-to-tau" class="section level3">
<h3>Final Step: Conversion from <code>omega</code> to
<code>tau</code></h3>
<p>After fitting the Bayesian model, the posterior probabilities
(<code>omega</code>) represent the contribution of theoretical clonal
peaks. We convert these probabilities into clonal timing estimates
(<code>tau</code>) using the following equations.</p>
<p>For <code>2:1</code> karyotypes:</p>
<p><span class="math display">\[
\tau = \frac{3\omega_2}{2\omega_2 + \omega_1}
\]</span></p>
<p>For <code>2:2</code> and <code>2:0</code> karyotypes:</p>
<p><span class="math display">\[
\tau = \frac{2\omega_2}{2\omega_2 + \omega_1}
\]</span></p>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
