<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index • tickTack</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta property="og:image" content="https://caravagnalab.github.io/tickTack/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tickTack</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/tickTack.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/a1_introduction.html">1. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/a2_Single_segment_timing.html">2. Timing Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a3_Understanding_single_segment_timing.html">3. Understanding Timing of Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a4_Multiple_segment_timing.html">4. Timing Clonal Peaks in a hierarchical fashioin</a></li>
    <li><a class="dropdown-item" href="../articles/a5_Understanding_multiple_segments_timing.html">5. Understanding Timing of Multiple Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a6_Simulate_data.html">6. Simulate data</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="CNA.html">CNA()</a></code>

        </dt>
        <dd>Extract CNAs.</dd>
      </dl><dl><dt>

          <code><a href="Mutations.html">Mutations()</a></code>

        </dt>
        <dd>Extract mutations.</dd>
      </dl><dl><dt>

          <code><a href="add_DP_and_NV.html">add_DP_and_NV()</a></code>

        </dt>
        <dd>add_DP_and_NV Function</dd>
      </dl><dl><dt>

          <code><a href="add_drivers_to_segment_plot.html">add_drivers_to_segment_plot()</a></code>

        </dt>
        <dd>Add driver mutation annotations to a segment plot</dd>
      </dl><dl><dt>

          <code><a href="chr_coordinates_GRCh38.html">chr_coordinates_GRCh38</a></code>

        </dt>
        <dd>GRCh38 coordinates</dd>
      </dl><dl><dt>

          <code><a href="chr_coordinates_hg19.html">chr_coordinates_hg19</a></code>

        </dt>
        <dd>hg19 coordinates</dd>
      </dl><dl><dt>

          <code><a href="convert_name.html">convert_name()</a></code>

        </dt>
        <dd>Convert model names to standardized names</dd>
      </dl><dl><dt>

          <code><a href="convert_to_granges.html">convert_to_granges()</a></code>

        </dt>
        <dd>convert_to_granges Function</dd>
      </dl><dl><dt>

          <code><a href="fit.html">fit()</a></code>

        </dt>
        <dd>Fit timing of clonal peaks in cancer genome sequencing data</dd>
      </dl><dl><dt>

          <code><a href="fit_h.html">fit_h()</a></code>

        </dt>
        <dd>Get the fit to obtain the clocks posteriors</dd>
      </dl><dl><dt>

          <code><a href="fit_h_single_k.html">fit_h_single_k()</a></code>

        </dt>
        <dd>Get the fit to obtain the clocks posteriors</dd>
      </dl><dl><dt>

          <code><a href="fit_variational_h.html">fit_variational_h()</a></code>

        </dt>
        <dd>fit_variational Function</dd>
      </dl><dl><dt>

          <code><a href="get_clonal_peaks.html">get_clonal_peaks()</a></code>

        </dt>
        <dd>get_clonal_peaks Function</dd>
      </dl><dl><dt>

          <code><a href="get_drivers.html">get_drivers()</a></code>

        </dt>
        <dd>Extract driver mutations from mutation data</dd>
      </dl><dl><dt>

          <code><a href="get_initialization.html">get_initialization()</a></code>

        </dt>
        <dd>get_initialization</dd>
      </dl><dl><dt>

          <code><a href="get_k_max_k_means.html">get_k_max_k_means()</a></code>

        </dt>
        <dd>model_selection Function</dd>
      </dl><dl><dt>

          <code><a href="get_major.html">get_major()</a></code>

        </dt>
        <dd>get_major Function</dd>
      </dl><dl><dt>

          <code><a href="get_reference.html">get_reference()</a></code>

        </dt>
        <dd>Get reference genome coordinates</dd>
      </dl><dl><dt>

          <code><a href="get_simulation.html">get_simulation()</a></code>

        </dt>
        <dd>get_simulation Function</dd>
      </dl><dl><dt>

          <code><a href="get_simulation_tickTack.html">get_simulation_tickTack()</a></code>

        </dt>
        <dd>get_taus_karyo Function</dd>
      </dl><dl><dt>

          <code><a href="get_taus_karyo.html">get_taus_karyo()</a></code>

        </dt>
        <dd>get_taus_karyo Function</dd>
      </dl><dl><dt>

          <code><a href="get_taus_karyo_gerstrung.html">get_taus_karyo_gerstrung()</a></code>

        </dt>
        <dd>get_taus_karyo gerstrung</dd>
      </dl><dl><dt>

          <code><a href="has_driver_data.html">has_driver_data()</a></code>

        </dt>
        <dd>Check if mutation data has driver annotations</dd>
      </dl><dl><dt>

          <code><a href="karyo_to_int.html">karyo_to_int()</a></code>

        </dt>
        <dd>karyo_to_int Function</dd>
      </dl><dl><dt>

          <code><a href="model_selection_h.html">model_selection_h()</a></code>

        </dt>
        <dd>model_selection Function</dd>
      </dl><dl><dt>

          <code><a href="pcawg_example.html">pcawg_example</a></code>

        </dt>
        <dd>pcawg_example</dd>
      </dl><dl><dt>

          <code><a href="pcawg_example_2.html">pcawg_example_2</a></code>

        </dt>
        <dd>pcawg_example_2</dd>
      </dl><dl><dt>

          <code><a href="plot_cnaqc.html">plot_cnaqc()</a></code>

        </dt>
        <dd>Plot</dd>
      </dl><dl><dt>

          <code><a href="plot_elbo_h.html">plot_elbo_h()</a></code>

        </dt>
        <dd>Plot elbo Function</dd>
      </dl><dl><dt>

          <code><a href="plot_model_selection_h.html">plot_model_selection_h()</a></code>

        </dt>
        <dd>Plot the behavior of the model selection scores vs the number of clusters</dd>
      </dl><dl><dt>

          <code><a href="plot_posterior_clocks_h.html">plot_posterior_clocks_h()</a></code>

        </dt>
        <dd>Plot posterior clocks distributions obtained from the hierarchical model fit</dd>
      </dl><dl><dt>

          <code><a href="plot_posterior_weights_h.html">plot_posterior_weights_h()</a></code>

        </dt>
        <dd>Plot posterior weights distributions obtained from the hierarchical model fit</dd>
      </dl><dl><dt>

          <code><a href="plot_segments_tick_tack.html">plot_segments_tick_tack()</a></code>

        </dt>
        <dd>Plot segmentation results from tickTack analysis</dd>
      </dl><dl><dt>

          <code><a href="plot_segments_tick_tack_CN.html">plot_segments_tick_tack_CN()</a></code>

        </dt>
        <dd>Plot copy number segments from CNAqc</dd>
      </dl><dl><dt>

          <code><a href="plot_segments_tick_tack_data.html">plot_segments_tick_tack_data()</a></code>

        </dt>
        <dd>Plot mutation data from tickTack analysis</dd>
      </dl><dl><dt>

          <code><a href="plot_timing.html">plot_timing()</a></code>

        </dt>
        <dd>Plot timing of clonal peaks in cancer genome sequencing data</dd>
      </dl><dl><dt>

          <code><a href="plot_timing_h.html">plot_timing_h()</a></code>

        </dt>
        <dd>plot_timing_h Function</dd>
      </dl><dl><dt>

          <code><a href="plot_vaf.html">plot_vaf()</a></code>

        </dt>
        <dd>Plot variant allele frequency (VAF) distributions</dd>
      </dl><dl><dt>

          <code><a href="prepare_input_data.html">prepare_input_data()</a></code>

        </dt>
        <dd>prepare_input_data Function</dd>
      </dl><dl><dt>

          <code><a href="simulate_mutations.html">simulate_mutations()</a></code>

        </dt>
        <dd>simulate_mutations Function</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

