# Package index

## All functions

- [`CNA()`](https://caravagnalab.github.io/tickTack/reference/CNA.md) :
  Extract CNAs.
- [`Mutations()`](https://caravagnalab.github.io/tickTack/reference/Mutations.md)
  : Extract mutations.
- [`add_DP_and_NV()`](https://caravagnalab.github.io/tickTack/reference/add_DP_and_NV.md)
  : add_DP_and_NV Function
- [`add_drivers_to_segment_plot()`](https://caravagnalab.github.io/tickTack/reference/add_drivers_to_segment_plot.md)
  : Add driver mutation annotations to a segment plot
- [`chr_coordinates_GRCh38`](https://caravagnalab.github.io/tickTack/reference/chr_coordinates_GRCh38.md)
  : GRCh38 coordinates
- [`chr_coordinates_hg19`](https://caravagnalab.github.io/tickTack/reference/chr_coordinates_hg19.md)
  : hg19 coordinates
- [`convert_name()`](https://caravagnalab.github.io/tickTack/reference/convert_name.md)
  : Convert model names to standardized names
- [`convert_to_granges()`](https://caravagnalab.github.io/tickTack/reference/convert_to_granges.md)
  : convert_to_granges Function
- [`fit()`](https://caravagnalab.github.io/tickTack/reference/fit.md) :
  Fit timing of clonal peaks in cancer genome sequencing data
- [`fit_h()`](https://caravagnalab.github.io/tickTack/reference/fit_h.md)
  : Get the fit to obtain the clocks posteriors
- [`fit_h_single_k()`](https://caravagnalab.github.io/tickTack/reference/fit_h_single_k.md)
  : Get the fit to obtain the clocks posteriors
- [`fit_variational_h()`](https://caravagnalab.github.io/tickTack/reference/fit_variational_h.md)
  : fit_variational Function
- [`get_clonal_peaks()`](https://caravagnalab.github.io/tickTack/reference/get_clonal_peaks.md)
  : get_clonal_peaks Function
- [`get_drivers()`](https://caravagnalab.github.io/tickTack/reference/get_drivers.md)
  : Extract driver mutations from mutation data
- [`get_initialization()`](https://caravagnalab.github.io/tickTack/reference/get_initialization.md)
  : get_initialization
- [`get_k_max_k_means()`](https://caravagnalab.github.io/tickTack/reference/get_k_max_k_means.md)
  : model_selection Function
- [`get_major()`](https://caravagnalab.github.io/tickTack/reference/get_major.md)
  : get_major Function
- [`get_reference()`](https://caravagnalab.github.io/tickTack/reference/get_reference.md)
  : Get reference genome coordinates
- [`get_simulation()`](https://caravagnalab.github.io/tickTack/reference/get_simulation.md)
  : get_simulation Function
- [`get_simulation_tickTack()`](https://caravagnalab.github.io/tickTack/reference/get_simulation_tickTack.md)
  : get_taus_karyo Function
- [`get_taus_karyo()`](https://caravagnalab.github.io/tickTack/reference/get_taus_karyo.md)
  : get_taus_karyo Function
- [`get_taus_karyo_gerstrung()`](https://caravagnalab.github.io/tickTack/reference/get_taus_karyo_gerstrung.md)
  : get_taus_karyo gerstrung
- [`has_driver_data()`](https://caravagnalab.github.io/tickTack/reference/has_driver_data.md)
  : Check if mutation data has driver annotations
- [`karyo_to_int()`](https://caravagnalab.github.io/tickTack/reference/karyo_to_int.md)
  : karyo_to_int Function
- [`model_selection_h()`](https://caravagnalab.github.io/tickTack/reference/model_selection_h.md)
  : model_selection Function
- [`pcawg_example`](https://caravagnalab.github.io/tickTack/reference/pcawg_example.md)
  : pcawg_example
- [`pcawg_example_2`](https://caravagnalab.github.io/tickTack/reference/pcawg_example_2.md)
  : pcawg_example_2
- [`plot_cnaqc()`](https://caravagnalab.github.io/tickTack/reference/plot_cnaqc.md)
  : Plot
- [`plot_cnaqc_choose_K()`](https://caravagnalab.github.io/tickTack/reference/plot_cnaqc_choose_K.md)
  : Plot cnaqc choosing K
- [`plot_elbo_h()`](https://caravagnalab.github.io/tickTack/reference/plot_elbo_h.md)
  : Plot elbo Function
- [`plot_model_selection_h()`](https://caravagnalab.github.io/tickTack/reference/plot_model_selection_h.md)
  : Plot the behavior of the model selection scores vs the number of
  clusters
- [`plot_posterior_clocks_h()`](https://caravagnalab.github.io/tickTack/reference/plot_posterior_clocks_h.md)
  : Plot posterior clocks distributions obtained from the hierarchical
  model fit
- [`plot_posterior_weights_h()`](https://caravagnalab.github.io/tickTack/reference/plot_posterior_weights_h.md)
  : Plot posterior weights distributions obtained from the hierarchical
  model fit
- [`plot_segments_tick_tack()`](https://caravagnalab.github.io/tickTack/reference/plot_segments_tick_tack.md)
  : Plot segmentation results from tickTack analysis
- [`plot_segments_tick_tack_CN()`](https://caravagnalab.github.io/tickTack/reference/plot_segments_tick_tack_CN.md)
  : Plot copy number segments from CNAqc
- [`plot_segments_tick_tack_data()`](https://caravagnalab.github.io/tickTack/reference/plot_segments_tick_tack_data.md)
  : Plot mutation data from tickTack analysis
- [`plot_timing()`](https://caravagnalab.github.io/tickTack/reference/plot_timing.md)
  : Plot timing of clonal peaks in cancer genome sequencing data
- [`plot_timing_h()`](https://caravagnalab.github.io/tickTack/reference/plot_timing_h.md)
  : plot_timing_h Function
- [`plot_vaf()`](https://caravagnalab.github.io/tickTack/reference/plot_vaf.md)
  : Plot variant allele frequency (VAF) distributions
- [`predict_best_number_of_clusters()`](https://caravagnalab.github.io/tickTack/reference/predict_best_number_of_clusters.md)
  : Model Saelection.
- [`prepare_input_data()`](https://caravagnalab.github.io/tickTack/reference/prepare_input_data.md)
  : prepare_input_data Function
- [`process_file_prediction()`](https://caravagnalab.github.io/tickTack/reference/process_file_prediction.md)
  : Model Selection function.
- [`simulate_mutations()`](https://caravagnalab.github.io/tickTack/reference/simulate_mutations.md)
  : simulate_mutations Function
