<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>3. Understanding Timing of Clonal Peaks • tickTack</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="3. Understanding Timing of Clonal Peaks">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tickTack</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/tickTack.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_introduction.html">1. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/a2_Single_segment_timing.html">2. Timing Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a3_Understanding_single_segment_timing.html">3. Understanding Timing of Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a4_Multiple_segment_timing.html">4. Timing Clonal Peaks in a hierarchical fashioin</a></li>
    <li><a class="dropdown-item" href="../articles/a5_Understanding_multiple_segments_timing.html">5. Understanding Timing of Multiple Clonal Peaks</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>3. Understanding Timing of Clonal Peaks</h1>
            
      

      <div class="d-none name"><code>a3_Understanding_single_segment_timing.Rmd</code></div>
    </div>

    
    
<p>This vignette delves into how the <code>tickTack</code> package
obtains timing results for clonal peaks using the <code>fit</code>
function. It explains the underlying processes, including the
computation of clonal peaks and the Stan model used for inference.</p>
<div class="section level2">
<h2 id="key-components-of-the-analysis">Key Components of the Analysis<a class="anchor" aria-label="anchor" href="#key-components-of-the-analysis"></a>
</h2>
<p>The timing results are obtained in the following steps:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Extracting clonal peaks</strong>: Theoretical peaks in the
Variant Allele Frequency (VAF) spectrum are calculated for specific
karyotypes and tumor purity.</li>
<li>
<strong>Filtering mutations</strong>: Mutations within confidence
intervals around these peaks are selected for analysis.</li>
<li>
<strong>Fitting the model</strong>: Bayesian inference is performed
using a Stan model to estimate the timing of clonal expansions.</li>
</ol>
<div class="section level3">
<h3 id="step-1-calculating-clonal-peaks">Step 1: Calculating Clonal Peaks<a class="anchor" aria-label="anchor" href="#step-1-calculating-clonal-peaks"></a>
</h3>
<p>The clonal peaks are calculated using the
<code>get_clonal_peaks</code> function. The function computes the
theoretical VAF peaks based on the karyotype and tumor purity. They are
computed using the following equation:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mtext mathvariant="normal">Peak</mtext><mi>i</mi></msub><mo>=</mo><mfrac><mrow><msub><mi>M</mi><mi>i</mi></msub><mo>⋅</mo><mi>P</mi></mrow><mrow><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>N</mi><mtext mathvariant="normal">tot</mtext></msub><mo>⋅</mo><mi>P</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mn>2</mn><mo>⋅</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>P</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac></mrow><annotation encoding="application/x-tex">
\text{Peak}_i = \frac{M_i \cdot P}{(N_{\text{tot}} \cdot P) + 2 \cdot (1 - P)}
</annotation></semantics></math></p>
<p>Where: -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>M</mi><mi>i</mi></msub><annotation encoding="application/x-tex">M_i</annotation></semantics></math>:
The multiplicity of the allele of interest (e.g., the major allele or a
single copy for heterozygosity). -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>P</mi><annotation encoding="application/x-tex">P</annotation></semantics></math>:
The tumor purity (a value between 0 and 1). -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>N</mi><mtext mathvariant="normal">tot</mtext></msub><annotation encoding="application/x-tex">N_{\text{tot}}</annotation></semantics></math>:
The total number of copies in the karyotype (e.g., sum of the major and
minor alleles). -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mn>2</mn><annotation encoding="application/x-tex">2</annotation></semantics></math>:
Accounts for the two copies in normal diploid cells.</p>
<div class="section level4">
<h4 id="explanation">Explanation<a class="anchor" aria-label="anchor" href="#explanation"></a>
</h4>
<ul>
<li>The <strong>numerator</strong>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>M</mi><mi>i</mi></msub><mo>⋅</mo><mi>P</mi></mrow><annotation encoding="application/x-tex">M_i \cdot P</annotation></semantics></math>,
represents the contribution of the tumor cell’s allele multiplicity
adjusted for tumor purity.</li>
<li>The <strong>denominator</strong>,
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>N</mi><mtext mathvariant="normal">tot</mtext></msub><mo>⋅</mo><mi>P</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mn>2</mn><mo>⋅</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>P</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">(N_{\text{tot}} \cdot P) + 2 \cdot (1 - P)</annotation></semantics></math>,
normalizes the peak to account for contributions from both tumor and
normal diploid cells in the sample.</li>
</ul>
</div>
<div class="section level4">
<h4 id="example-calculation">Example Calculation<a class="anchor" aria-label="anchor" href="#example-calculation"></a>
</h4>
<p>For a karyotype <span class="math inline">$k = "2:1"$</span> (major
allele = 2, minor allele = 1) and tumor purity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>P</mi><mo>=</mo><mn>0.4</mn></mrow><annotation encoding="application/x-tex">P = 0.4</annotation></semantics></math>:</p>
<ol style="list-style-type: decimal">
<li>
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>N</mi><mtext mathvariant="normal">tot</mtext></msub><mo>=</mo><mn>2</mn><mo>+</mo><mn>1</mn><mo>=</mo><mn>3</mn></mrow><annotation encoding="application/x-tex">N_{\text{tot}} = 2 + 1 = 3</annotation></semantics></math>.</li>
<li>Compute peaks for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>M</mi><mi>i</mi></msub><mo>=</mo><mn>1</mn></mrow><annotation encoding="application/x-tex">M_i = 1</annotation></semantics></math>
(single allele) and
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>M</mi><mi>i</mi></msub><mo>=</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">M_i = 2</annotation></semantics></math>
(major allele):
<math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mtext mathvariant="normal">Peak</mtext><mn>1</mn></msub><mo>=</mo><mfrac><mrow><mn>1</mn><mo>⋅</mo><mn>0.4</mn></mrow><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo>⋅</mo><mn>0.4</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mn>2</mn><mo>⋅</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mn>0.4</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><mo>=</mo><mfrac><mn>0.4</mn><mrow><mn>1.2</mn><mo>+</mo><mn>1.2</mn></mrow></mfrac><mo>=</mo><mn>0.1667</mn></mrow><annotation encoding="application/x-tex">
\text{Peak}_1 = \frac{1 \cdot 0.4}{(3 \cdot 0.4) + 2 \cdot (1 - 0.4)} = \frac{0.4}{1.2 + 1.2} = 0.1667
</annotation></semantics></math><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mtext mathvariant="normal">Peak</mtext><mn>2</mn></msub><mo>=</mo><mfrac><mrow><mn>2</mn><mo>⋅</mo><mn>0.4</mn></mrow><mrow><mrow><mo stretchy="true" form="prefix">(</mo><mn>3</mn><mo>⋅</mo><mn>0.4</mn><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mn>2</mn><mo>⋅</mo><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mn>0.4</mn><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><mo>=</mo><mfrac><mn>0.8</mn><mrow><mn>1.2</mn><mo>+</mo><mn>1.2</mn></mrow></mfrac><mo>=</mo><mn>0.3333</mn></mrow><annotation encoding="application/x-tex">
\text{Peak}_2 = \frac{2 \cdot 0.4}{(3 \cdot 0.4) + 2 \cdot (1 - 0.4)} = \frac{0.8}{1.2 + 1.2} = 0.3333
</annotation></semantics></math>
</li>
</ol>
</div>
</div>
<div class="section level3">
<h3 id="step-2-filtering-mutations">Step 2: Filtering Mutations<a class="anchor" aria-label="anchor" href="#step-2-filtering-mutations"></a>
</h3>
<p>Once the peaks are obtained, mutations within a genomic segment are
filtered based on whether their VAF values fall within a confidence
interval around these peaks.</p>
<p>The confidence intervals are computed using either:</p>
<ul>
<li>
<strong>Binomial distribution</strong>: for standard VAF
estimation</li>
<li>
<strong>Beta-binomial distribution</strong>: to account for
overdispersion</li>
</ul>
<div class="section level4">
<h4 id="filtering-logic">Filtering logic<a class="anchor" aria-label="anchor" href="#filtering-logic"></a>
</h4>
<p>For each mutation, the number of variant reads (NV) is compared to
the confidence interval calculated for the depth (DP) at each peak:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example pseudo-code for filtering</span></span>
<span><span class="va">probs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">alpha</span> <span class="op">/</span> <span class="fl">2</span>, <span class="fl">1</span> <span class="op">-</span> <span class="va">alpha</span> <span class="op">/</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">p</span> <span class="kw">in</span> <span class="va">peaks</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">quantiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Binomial.html" class="external-link">qbinom</a></span><span class="op">(</span><span class="va">probs</span>, <span class="va">DP</span>, <span class="va">p</span><span class="op">)</span> <span class="co"># or qbb for beta-binomial</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="va">NV</span> <span class="op">&gt;=</span> <span class="va">quantiles</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&amp;&amp;</span> <span class="va">NV</span> <span class="op">&lt;=</span> <span class="va">quantiles</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># Mutation is accepted</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Only mutations passing this test are used for model fitting.</p>
</div>
</div>
<div class="section level3">
<h3 id="step-3-bayesian-inference-with-stan">Step 3: Bayesian Inference with Stan<a class="anchor" aria-label="anchor" href="#step-3-bayesian-inference-with-stan"></a>
</h3>
<p>The final step is to fit a Stan model to the filtered mutations. The
model estimates the posterior distribution of <code>tau</code>, which
represents the timing of clonal expansions.</p>
<p>The stan code used for the model is divided in the following
pieces.</p>
<div class="section level4">
<h4 id="data-block">Data block<a class="anchor" aria-label="anchor" href="#data-block"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>data {</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  int N;</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  array[N] int NV;</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  array[N] int DP;</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  array[<span class="dv">2</span>] real peaks;</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>}</span></code></pre></div>
<p>Here, the data block defines inputs that the model uses:</p>
<ul>
<li>
<strong>N</strong>: Total number of mutations to analyze.</li>
<li>
<strong>NV</strong>: Number of variant reads observed for each
mutation.</li>
<li>
<strong>DP</strong>: Total sequencing depth for each mutation.</li>
<li>
<strong>peaks</strong>: An array of two real values representing the
theoretical clonal peaks (e.g., 2:1 and 2:2 ratios or other hypothesized
clonal frequencies).</li>
</ul>
</div>
<div class="section level4">
<h4 id="parameters-block">Parameters block<a class="anchor" aria-label="anchor" href="#parameters-block"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>parameters {</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  simplex[<span class="dv">2</span>] omega;</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>}</span></code></pre></div>
<p>The parameters block contains:</p>
<ul>
<li>
<strong>omega</strong>: A vector representing the probability
contributions of each theoretical peak. It follows a Dirichlet
distribution (a standard approach for modeling probabilities that sum to
1).</li>
</ul>
</div>
<div class="section level4">
<h4 id="model-block">Model block<a class="anchor" aria-label="anchor" href="#model-block"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>model {</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  vector[<span class="dv">2</span>] contributions;</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  omega <span class="sc">~</span> <span class="fu">dirichlet</span>(<span class="fu">rep_vector</span>(<span class="fl">2.0</span>, <span class="dv">2</span>));</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  </span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N) {</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>    <span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) {</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>      contributions[k] <span class="ot">=</span> <span class="fu">log</span>(omega[k]) <span class="sc">+</span> <span class="fu">binomial_lpmf</span>(NV[i] <span class="sc">|</span> DP[i], peaks[k]);</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>    }</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>    target <span class="sc">+</span><span class="er">=</span> <span class="fu">log_sum_exp</span>(contributions);</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  }</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>}</span></code></pre></div>
<p>The <code>model</code> block contains three components:</p>
<ul>
<li>
<strong>Prior</strong>: The line
<code>omega ~ dirichlet(rep_vector(2.0, 2));</code> applies a uniform
Dirichlet prior to the karyotype mixture probabilities (omega). This
means that we do not a priori favor one clonal peak over the other
unless there is data evidence.</li>
<li>
<strong>Likelihood</strong>: The likelihood is computed using the
<code>binomial_lpmf(NV[i] | DP[i], peaks[k])</code> (or the Negative
binomial). This term computes the probability of observing
<code>NV[i]</code> variant reads given sequencing depth
<code>DP[i]</code> and a given theoretical clonal peak value
<code>peaks[k]</code>.</li>
<li>
<strong>Marginalization</strong>: The line
<code>log_sum_exp(contributions)</code> computes the log marginal
likelihood by summing over possible contributions (karyotypes) using the
softmax approach, allowing the model to integrate over possible
karyotype contributions (<code>omega</code>).</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="final-step-conversion-from-omega-to-tau">Final Step: Conversion from <code>omega</code> to
<code>tau</code><a class="anchor" aria-label="anchor" href="#final-step-conversion-from-omega-to-tau"></a>
</h3>
<p>After fitting the Bayesian model, the posterior probabilities
(<code>omega</code>) represent the contribution of theoretical clonal
peaks. We convert these probabilities into clonal timing estimates
(<code>tau</code>) using the following equations.</p>
<p>For <code>2:1</code> karyotypes:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>τ</mi><mo>=</mo><mfrac><mrow><mn>3</mn><msub><mi>ω</mi><mn>2</mn></msub></mrow><mrow><mn>2</mn><msub><mi>ω</mi><mn>2</mn></msub><mo>+</mo><msub><mi>ω</mi><mn>1</mn></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">
\tau = \frac{3\omega_2}{2\omega_2 + \omega_1}
</annotation></semantics></math></p>
<p>For <code>2:2</code> and <code>2:0</code> karyotypes:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>τ</mi><mo>=</mo><mfrac><mrow><mn>2</mn><msub><mi>ω</mi><mn>2</mn></msub></mrow><mrow><mn>2</mn><msub><mi>ω</mi><mn>2</mn></msub><mo>+</mo><msub><mi>ω</mi><mn>1</mn></msub></mrow></mfrac></mrow><annotation encoding="application/x-tex">
\tau = \frac{2\omega_2}{2\omega_2 + \omega_1}
</annotation></semantics></math></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
