<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>4. Timing Clonal Peaks in a hierarchical fashioin â€¢ tickTack</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="4. Timing Clonal Peaks in a hierarchical fashioin">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tickTack</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/tickTack.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_introduction.html">1. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/a2_Single_segment_timing.html">2. Timing Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a3_Understanding_single_segment_timing.html">3. Understanding Timing of Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a4_Multiple_segment_timing.html">4. Timing Clonal Peaks in a hierarchical fashioin</a></li>
    <li><a class="dropdown-item" href="../articles/a5_Understanding_multiple_segments_timing.html">5. Understanding Timing of Multiple Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a6_Simulate_data.html">6. Simulate data</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>4. Timing Clonal Peaks in a hierarchical fashioin</h1>
            
      

      <div class="d-none name"><code>a4_Multiple_segment_timing.Rmd</code></div>
    </div>

    
    
<p>The <code>_fit_h</code> function in the <code>tickTack</code> package
estimates the timing of <code>K</code> clonal peaks in cancer genome
sequencing data according to which a clustering can be performed. This
vignette describes the functionality of the <code>fit_h</code> function,
including input requirements, output, and an example analysis using the
<code><a href="../reference/pcawg_example_2.html">tickTack::pcawg_example_2</a></code> dataset.</p>
<div class="section level2">
<h2 id="overview-of-the-fit_h-function">Overview of the <code>fit_h</code> Function<a class="anchor" aria-label="anchor" href="#overview-of-the-fit_h-function"></a>
</h2>
<p>The <code>fit_h</code> function uses a hierarchical model to fit
clonal peaks in sequencing data considering the grouping tructure of the
chromosomes segments. It identifies segments of the genome with specific
karyotypes and mutations that meet the input criteria, then estimates
the timing of the groups of events and assign each segment to a
clock.</p>
<div class="section level3">
<h3 id="key-parameters">Key Parameters<a class="anchor" aria-label="anchor" href="#key-parameters"></a>
</h3>
<ul>
<li>
<strong><code>x</code></strong>: a CNAqc object with mutations, cna
and metadata</li>
<li>
<strong><code>max_attempts</code></strong>: Number of times the
variational inference is repeated to avoid local minima.</li>
<li>
<strong><code>INIT</code></strong>: Logical flag to pass some
initialization values tothe variational inference, default is
<code>TRURE</code>.</li>
<li>
<strong><code>tolerance</code></strong>: tolerance between two value
of subsequent iterations of gradient ascent pn elbo, default is
<code>0.01</code>.</li>
<li>
<strong><code>possible_k</code></strong>: A character vector of
possible karyotypes, defaulting to
<code>c("2:1", "2:2", "2:0")</code>.</li>
<li>
<strong><code>alpha</code></strong>: Significance level, defaulting
to <code>0.05</code>.</li>
<li>
<strong><code>min_mutations_number</code></strong>: Minimum number
of mutations required for analysis, defaulting to <code>2</code>.</li>
<li>
<strong><code>n_components</code></strong>: If <code>0</code>, then
the strategy to choose the #components follows the default procedure,
otherwise the inference is repeated for K equal up to a maximun of
n_components.</li>
</ul>
</div>
<div class="section level3">
<h3 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h3>
<p>The function returns a list containing:</p>
<ol style="list-style-type: decimal">
<li>
<strong><code>data</code></strong>: The data used to perform the
inference after selecting the ones that respect the assumptions to be
used in the model.</li>
<li>
<strong><code>draws_and_summary</code></strong>: List of 3 for each
K the inference is performed with. draws are available both for the
clocks and for the weights Summary statistics for the estimated timing
of clonal peaks.</li>
<li>
<strong><code>log_lik_matrix_list</code></strong>: Summary
statistics for the estimated timing of clonal peaks.</li>
<li>
<strong><code>elbo_iterations</code></strong>: Summary statistics
for the estimated timing of clonal peaks.</li>
</ol>
<p>If no segments meet the criteria, the function returns
<code>NULL</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="analyzing-ticktackpcawg_example_2">Analyzing <code>tickTack::pcawg_example_2</code><a class="anchor" aria-label="anchor" href="#analyzing-ticktackpcawg_example_2"></a>
</h2>
<p>We will use the <code><a href="../reference/pcawg_example_2.html">tickTack::pcawg_example_2</a></code> dataset to
demonstrate how to use the <code>fit_h</code> function.</p>
<div class="section level3">
<h3 id="input-data">Input Data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h3>
<p>The <code><a href="../reference/pcawg_example_2.html">tickTack::pcawg_example_2</a></code> dataset contains three
components:</p>
<ul>
<li>
<strong><code>mutations</code></strong>: Mutation data.</li>
<li>
<strong><code>cna</code></strong>: Copy number alterations
(CNA).</li>
<li>
<strong><code>metadata</code></strong>: Sample metadata, including
tumor purity.</li>
</ul>
<p>Preview the data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://caravagnalab.github.io/tickTack/">tickTack</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># View example dataset components</span></span>
<span><span class="va">mutations</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_example_2.html">pcawg_example_2</a></span><span class="op">$</span><span class="va">mutations</span></span>
<span><span class="va">cna</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_example_2.html">pcawg_example_2</a></span><span class="op">$</span><span class="va">cna</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_example_2.html">pcawg_example_2</a></span><span class="op">$</span><span class="va">metadata</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mutations</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    chr    from      to ref alt DP NV       VAF</span></span>
<span><span class="co">## 1 chr1 1018754 1018754   C   C 51 24 0.4705882</span></span>
<span><span class="co">## 2 chr1 1107556 1107556   C   C 53 17 0.3207547</span></span>
<span><span class="co">## 3 chr1 1127192 1127192   C   C 78 13 0.1666667</span></span>
<span><span class="co">## 4 chr1 1255263 1255263   C   C 66 23 0.3484848</span></span>
<span><span class="co">## 5 chr1 1474126 1474126   C   C 68 17 0.2500000</span></span>
<span><span class="co">## 6 chr1 1532289 1532289   G   G 39  6 0.1538462</span></span>
<span><span class="co">##                                 sample NR Hugo_Symbol Strand</span></span>
<span><span class="co">## 1 3b7810f7-f8ff-4d62-b766-3ba06170194c 27    C1orf159      +</span></span>
<span><span class="co">## 2 3b7810f7-f8ff-4d62-b766-3ba06170194c 36      TTLL10      +</span></span>
<span><span class="co">## 3 3b7810f7-f8ff-4d62-b766-3ba06170194c 65      TTLL10      +</span></span>
<span><span class="co">## 4 3b7810f7-f8ff-4d62-b766-3ba06170194c 43      CPSF3L      +</span></span>
<span><span class="co">## 5 3b7810f7-f8ff-4d62-b766-3ba06170194c 51     TMEM240      +</span></span>
<span><span class="co">## 6 3b7810f7-f8ff-4d62-b766-3ba06170194c 33    C1orf233      +</span></span>
<span><span class="co">##   Variant_Classification Variant_Type Tumor_Seq_Allele2    dbSNP_RS</span></span>
<span><span class="co">## 1                 Intron          SNP                 T rs188930619</span></span>
<span><span class="co">## 2                5'Flank          SNP                 T        &lt;NA&gt;</span></span>
<span><span class="co">## 3                 Intron          SNP                 T        &lt;NA&gt;</span></span>
<span><span class="co">## 4                 Intron          SNP                 T rs552646507</span></span>
<span><span class="co">## 5                 Intron          SNP                 T rs572658610</span></span>
<span><span class="co">## 6                    IGR          SNP                 A rs575347755</span></span>
<span><span class="co">##   dbSNP_Val_Status          Matched_Norm_Sample_Barcode     Genome_Change</span></span>
<span><span class="co">## 1             &lt;NA&gt; 5d5a6488-dc29-4ab5-8a98-e94fd5be113f g.chr1:1018754C&gt;T</span></span>
<span><span class="co">## 2             &lt;NA&gt; 5d5a6488-dc29-4ab5-8a98-e94fd5be113f g.chr1:1107556C&gt;T</span></span>
<span><span class="co">## 3             &lt;NA&gt; 5d5a6488-dc29-4ab5-8a98-e94fd5be113f g.chr1:1127192C&gt;T</span></span>
<span><span class="co">## 4      byFrequency 5d5a6488-dc29-4ab5-8a98-e94fd5be113f g.chr1:1255263C&gt;T</span></span>
<span><span class="co">## 5      byFrequency 5d5a6488-dc29-4ab5-8a98-e94fd5be113f g.chr1:1474126C&gt;T</span></span>
<span><span class="co">## 6             &lt;NA&gt; 5d5a6488-dc29-4ab5-8a98-e94fd5be113f g.chr1:1532289G&gt;A</span></span>
<span><span class="co">##             ref_context gc_content i_1000genomes_AF i_1000genomes_ID</span></span>
<span><span class="co">## 1 cagGGGGTTACGTCAGGGTCT      0.602       0.00019968      rs188930619</span></span>
<span><span class="co">## 2 ccagctcaggcatcagtggct      0.692               NA             &lt;NA&gt;</span></span>
<span><span class="co">## 3 gtatcccttacgggaaacgaa      0.418               NA             &lt;NA&gt;</span></span>
<span><span class="co">## 4 gCAAGGGCAGCGCCTGGCAAG      0.652       0.00099840      rs552646507</span></span>
<span><span class="co">## 5 GCATGTCACACGCTTAGCAAC      0.587               NA             &lt;NA&gt;</span></span>
<span><span class="co">## 6 ACTTCGGGGCGGGACTGGAAG      0.607               NA             &lt;NA&gt;</span></span>
<span><span class="co">##                i_Callers i_GERM1000G i_GERMOVLP i_LOWSUPPORT i_NORMALPANEL</span></span>
<span><span class="co">## 1 broad,dkfz,muse,sanger          NA         NA           NA            NA</span></span>
<span><span class="co">## 2 broad,dkfz,muse,sanger          NA         NA           NA            NA</span></span>
<span><span class="co">## 3 broad,dkfz,muse,sanger          NA         NA           NA            NA</span></span>
<span><span class="co">## 4 broad,dkfz,muse,sanger          NA         NA           NA            NA</span></span>
<span><span class="co">## 5 broad,dkfz,muse,sanger          NA         NA           NA            NA</span></span>
<span><span class="co">## 6 broad,dkfz,muse,sanger          NA         NA           NA            NA</span></span>
<span><span class="co">##   i_NumCallers i_OXOGFAIL i_REMAPFAIL i_SEXF  i_VAF i_bPcr i_bSeq i_qual</span></span>
<span><span class="co">## 1            4         NA          NA     NA 0.4706     NA     NA   None</span></span>
<span><span class="co">## 2            4         NA          NA     NA 0.3208     NA     NA   None</span></span>
<span><span class="co">## 3            4         NA          NA     NA 0.1667     NA     NA   None</span></span>
<span><span class="co">## 4            4         NA          NA     NA 0.3485     NA     NA   None</span></span>
<span><span class="co">## 5            4         NA          NA     NA   0.25     NA     NA   None</span></span>
<span><span class="co">## 6            4         NA          NA     NA 0.1538     NA     NA   None</span></span>
<span><span class="co">##   i_repeat_masker i_signature_N3 i_signature_R1 i_signature_R2 i_snv_near_indel</span></span>
<span><span class="co">## 1            &lt;NA&gt;          FALSE          FALSE          FALSE            FALSE</span></span>
<span><span class="co">## 2           MLT1D          FALSE          FALSE          FALSE            FALSE</span></span>
<span><span class="co">## 3          MER9a1          FALSE          FALSE          FALSE            FALSE</span></span>
<span><span class="co">## 4            &lt;NA&gt;          FALSE          FALSE          FALSE            FALSE</span></span>
<span><span class="co">## 5            &lt;NA&gt;          FALSE          FALSE          FALSE            FALSE</span></span>
<span><span class="co">## 6            &lt;NA&gt;          FALSE          FALSE          FALSE            FALSE</span></span>
<span><span class="co">##   i_model_score i_n_vaf Project_Code Donor_ID</span></span>
<span><span class="co">## 1            NA      NA   Kidney-RCC  DO46877</span></span>
<span><span class="co">## 2            NA      NA   Kidney-RCC  DO46877</span></span>
<span><span class="co">## 3            NA      NA   Kidney-RCC  DO46877</span></span>
<span><span class="co">## 4            NA      NA   Kidney-RCC  DO46877</span></span>
<span><span class="co">## 5            NA      NA   Kidney-RCC  DO46877</span></span>
<span><span class="co">## 6            NA      NA   Kidney-RCC  DO46877</span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cna</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    chr      from        to Major minor CCF total_cn star level methods_agree</span></span>
<span><span class="co">## 1 chr1     10001 121499999     1     1   1        2    3     a             6</span></span>
<span><span class="co">## 2 chr1 121500000 128899999     1     1   1        2    2     d             4</span></span>
<span><span class="co">## 3 chr1 128900000 247247500     1     1   1        2    3     a             6</span></span>
<span><span class="co">## 4 chr1 247247501 249250620     1     1   1        2    2     d             4</span></span>
<span><span class="co">## 5 chr2     66017  90499999     1     1   1        2    3     a             6</span></span>
<span><span class="co">## 6 chr2  90500000  96799999     1     1   1        2    3     a             6</span></span>
<span><span class="co">##   absolute_broad_major_cn absolute_broad_minor_cn absolute_broad_het_error</span></span>
<span><span class="co">## 1                       1                       1               0.03993165</span></span>
<span><span class="co">## 2                       1                       1               0.05000000</span></span>
<span><span class="co">## 3                       1                       1               0.04043537</span></span>
<span><span class="co">## 4                      NA                      NA                       NA</span></span>
<span><span class="co">## 5                       1                       1               0.04002173</span></span>
<span><span class="co">## 6                       1                       1               0.05000000</span></span>
<span><span class="co">##   absolute_broad_cov_error aceseq_copy_number aceseq_minor_cn aceseq_major_cn</span></span>
<span><span class="co">## 1                0.1174128           2.000000        1.000000        1.000000</span></span>
<span><span class="co">## 2                0.1000000           2.000000        1.000000        1.000000</span></span>
<span><span class="co">## 3                0.1258711           2.000000        1.000000        1.000000</span></span>
<span><span class="co">## 4                       NA           2.136427        1.068214        1.068214</span></span>
<span><span class="co">## 5                0.1149430           2.000000        1.000000        1.000000</span></span>
<span><span class="co">## 6                0.3640664           2.000000        1.000000        1.000000</span></span>
<span><span class="co">##   battenberg_nMaj1_A battenberg_nMin1_A battenberg_frac1_A battenberg_nMaj2_A</span></span>
<span><span class="co">## 1                  1                  1                  1                 NA</span></span>
<span><span class="co">## 2                  1                  1                  1                 NA</span></span>
<span><span class="co">## 3                  1                  1                  1                 NA</span></span>
<span><span class="co">## 4                  1                  1                  1                 NA</span></span>
<span><span class="co">## 5                  1                  1                  1                 NA</span></span>
<span><span class="co">## 6                  1                  1                  1                 NA</span></span>
<span><span class="co">##   battenberg_nMin2_A battenberg_frac2_A battenberg_SDfrac_A</span></span>
<span><span class="co">## 1                 NA                 NA                  NA</span></span>
<span><span class="co">## 2                 NA                 NA                  NA</span></span>
<span><span class="co">## 3                 NA                 NA                  NA</span></span>
<span><span class="co">## 4                 NA                 NA                  NA</span></span>
<span><span class="co">## 5                 NA                 NA                  NA</span></span>
<span><span class="co">## 6                 NA                 NA                  NA</span></span>
<span><span class="co">##   battenberg_SDfrac_A_BS battenberg_frac1_A_0.025 battenberg_frac1_A_0.975</span></span>
<span><span class="co">## 1                     NA                       NA                       NA</span></span>
<span><span class="co">## 2                     NA                       NA                       NA</span></span>
<span><span class="co">## 3                     NA                       NA                       NA</span></span>
<span><span class="co">## 4                     NA                       NA                       NA</span></span>
<span><span class="co">## 5                     NA                       NA                       NA</span></span>
<span><span class="co">## 6                     NA                       NA                       NA</span></span>
<span><span class="co">##   clonehd_copy_number clonehd_minor_cn clonehd_major_cn sclust_nMaj1_A</span></span>
<span><span class="co">## 1                   2                1                1              1</span></span>
<span><span class="co">## 2                  NA               NA               NA             NA</span></span>
<span><span class="co">## 3                   2                1                1              1</span></span>
<span><span class="co">## 4                   2                1                1              1</span></span>
<span><span class="co">## 5                   2                1                1              1</span></span>
<span><span class="co">## 6                   2                1                1              1</span></span>
<span><span class="co">##   sclust_nMin1_A sclust_frac1_A sclust_nMaj2_A sclust_nMin2_A sclust_frac2_A</span></span>
<span><span class="co">## 1              1              1             NA             NA             NA</span></span>
<span><span class="co">## 2             NA             NA             NA             NA             NA</span></span>
<span><span class="co">## 3              1              1             NA             NA             NA</span></span>
<span><span class="co">## 4              1              1             NA             NA             NA</span></span>
<span><span class="co">## 5              1              1             NA             NA             NA</span></span>
<span><span class="co">## 6              1              1             NA             NA             NA</span></span>
<span><span class="co">##   jabba_copy_number jabba_minor_cn jabba_major_cn</span></span>
<span><span class="co">## 1                 2              1              1</span></span>
<span><span class="co">## 2                 2              1              1</span></span>
<span><span class="co">## 3                 2              1              1</span></span>
<span><span class="co">## 4                 2              1              1</span></span>
<span><span class="co">## 5                 2              1              1</span></span>
<span><span class="co">## 6                 2              1              1</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span></span></code></pre></div>
<pre><code><span><span class="co">##                                 sample purity ploidy purity_conf_mad wgd_status</span></span>
<span><span class="co">## 1 3b7810f7-f8ff-4d62-b766-3ba06170194c  0.611  1.924           0.002     no_wgd</span></span>
<span><span class="co">##   wgd_uncertain</span></span>
<span><span class="co">## 1         FALSE</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="running-the-fit_h-function">Running the <code>fit_h</code> function<a class="anchor" aria-label="anchor" href="#running-the-fit_h-function"></a>
</h3>
<p>We can run the <code>fit_h</code> function on the
<code><a href="../reference/pcawg_example_2.html">tickTack::pcawg_example_2</a></code> data to infer the timing of
clonal peaks</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract input data</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_example_2.html">pcawg_example_2</a></span></span>
<span><span class="va">tolerance</span> <span class="op">=</span> <span class="fl">0.1</span></span>
<span></span>
<span><span class="co"># Run the fit function</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_h.html">fit_h</a></span><span class="op">(</span></span>
<span>  x <span class="op">=</span> <span class="va">data</span>,</span>
<span>  max_attempts <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  INIT <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  tolerance <span class="op">=</span> <span class="va">tolerance</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="results">Results<a class="anchor" aria-label="anchor" href="#results"></a>
</h3>
<p>The <code>results</code> object that is returned together with the
CNAqc input object contains four components: <code>data</code>,
<code>draws_and_summary</code>, <code>log_lik_matrix_list</code> and
<code>elbo_iterations</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View summary for a specific K, here K = 2</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">results</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="interpreting-the-output">Interpreting the output<a class="anchor" aria-label="anchor" href="#interpreting-the-output"></a>
</h3>
<p>We can inspect the main output of interest to understand the timing
of clonal peaks. <code>results$draws_and_summary</code> contains: -
<strong><code>draws</code></strong> the draws from the approximate
posterior distribution of the taus and weights; -
<strong><code>summary</code></strong> a summary with the main statistics
of the approximate posterior distributions; -
<strong><code>summarized_results</code></strong> represents the clock
assignment, a tibble with the estimate of taus for each segment with a
copy number event that has been included in the hierarchical
inference</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View summary for a specific K, here K = 2</span></span>
<span><span class="va">results</span><span class="op">$</span><span class="va">draws_and_summary</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">summary</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 24 Ã— 7</span></span></span>
<span><span class="co">##    variable     mean   median       sd      mad         q5     q95</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> tau[1]   0.140    0.135    0.040<span style="text-decoration: underline;">6</span>   0.038<span style="text-decoration: underline;">9</span>   0.079<span style="text-decoration: underline;">9</span>     0.214  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> tau[2]   0.952    0.955    0.016<span style="text-decoration: underline;">9</span>   0.014<span style="text-decoration: underline;">4</span>   0.921      0.973  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> w[1,1]   0.844    0.940    0.215    0.080<span style="text-decoration: underline;">6</span>   0.299      0.998  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> w[2,1]   0.960    0.969    0.029<span style="text-decoration: underline;">9</span>   0.019<span style="text-decoration: underline;">8</span>   0.897      0.989  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> w[3,1]   0.988    0.990    0.007<span style="text-decoration: underline;">80</span>  0.005<span style="text-decoration: underline;">80</span>  0.973      0.996  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> w[4,1]   0.996    0.997    0.002<span style="text-decoration: underline;">48</span>  0.001<span style="text-decoration: underline;">89</span>  0.991      0.999  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> w[5,1]   0.000<span style="text-decoration: underline;">802</span> 0.000<span style="text-decoration: underline;">567</span> 0.000<span style="text-decoration: underline;">814</span> 0.000<span style="text-decoration: underline;">449</span> 0.000<span style="text-decoration: underline;">137</span>   0.002<span style="text-decoration: underline;">33</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> w[6,1]   0.009<span style="text-decoration: underline;">60</span>  0.000<span style="text-decoration: underline;">444</span> 0.040<span style="text-decoration: underline;">4</span>   0.000<span style="text-decoration: underline;">632</span> 0.000<span style="text-decoration: underline;">005</span>20 0.032<span style="text-decoration: underline;">8</span> </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> w[7,1]   0.037<span style="text-decoration: underline;">6</span>   0.001<span style="text-decoration: underline;">21</span>  0.121    0.001<span style="text-decoration: underline;">78</span>  0.000<span style="text-decoration: underline;">004</span>94 0.214  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> w[8,1]   0.979    0.985    0.022<span style="text-decoration: underline;">6</span>   0.011<span style="text-decoration: underline;">8</span>   0.940      0.996  </span></span>
<span><span class="co">## <span style="color: #949494;"># â„¹ 14 more rows</span></span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View detailed summarized results for a specific K, here K = 2</span></span>
<span><span class="va">results</span><span class="op">$</span><span class="va">draws_and_summary</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">summarized_results</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 11 Ã— 8</span></span></span>
<span><span class="co">##    segment_original_indx segment_name      segment_id karyotype chr   clock_mean</span></span>
<span><span class="co">##                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>                    38 chr5_75225353_75â€¦          1 2:1       chr5       0.135</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>                    40 chr5_76857368_76â€¦          2 2:1       chr5       0.135</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>                    42 chr5_77822720_82â€¦          3 2:1       chr5       0.135</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>                    44 chr5_83468190_18â€¦          4 2:1       chr5       0.135</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>                    46 chr6_242501_4335â€¦          5 2:0       chr6       0.955</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>                    48 chr6_4347801_290â€¦          6 2:0       chr6       0.955</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>                    50 chr6_29161839_38â€¦          7 2:0       chr6       0.955</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>                    54 chr7_65454_99371â€¦          8 2:1       chr7       0.135</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>                    56 chr7_9946564_579â€¦          9 2:1       chr7       0.135</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>                    57 chr7_5.8e+07_616â€¦         10 2:1       chr7       0.135</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">11</span>                    58 chr7_61700000_15â€¦         11 2:1       chr7       0.135</span></span>
<span><span class="co">## <span style="color: #949494;"># â„¹ 2 more variables: clock_low &lt;dbl&gt;, clock_high &lt;dbl&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="obtain-the-best-k-with-model_selection_h">Obtain the best K with model_selection_h<a class="anchor" aria-label="anchor" href="#obtain-the-best-k-with-model_selection_h"></a>
</h3>
<p>W e can run the <code>model_selection_h</code> function to obtain the
scores for each inference performed with a different K and take the one
with best ICL score if the BIC score prefer 2 components instead of 1,
otherwise choose 1 as best K. The function takes as input the
<code>results</code> and <code>n_components</code> and outputs the
<code>best_K</code> and the corresponding <code>best_fit</code> together
with the <code>model_selection_tibble</code> and the
<code>entropy_list</code> used to evaluate the ICL score.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results_model_selection</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/model_selection_h.html">model_selection_h</a></span><span class="op">(</span><span class="va">results</span>, n_components <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span>
<span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span>
<span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">best_K</span> <span class="op">&lt;-</span> <span class="va">results_model_selection</span><span class="op">$</span><span class="va">best_K</span></span>
<span><span class="va">model_selection_tibble</span> <span class="op">&lt;-</span> <span class="va">results_model_selection</span><span class="op">$</span><span class="va">model_selection_tibble</span></span>
<span><span class="va">entropy</span> <span class="op">&lt;-</span> <span class="va">results_model_selection</span><span class="op">$</span><span class="va">entropy_list</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visulizing-the-output">Visulizing the output<a class="anchor" aria-label="anchor" href="#visulizing-the-output"></a>
</h2>
<p>The results can be viewed is genome-wise perspective using the
<code><a href="../reference/plot_timing_h.html">tickTack::plot_timing_h</a></code> function.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_timing_h.html">plot_timing_h</a></span><span class="op">(</span><span class="va">results</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="a4_Multiple_segment_timing_files/figure-html/unnamed-chunk-6-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="visualize-distributions-of-draws-from-the-approximate-posterior">Visualize distributions of draws from the approximate posterior<a class="anchor" aria-label="anchor" href="#visualize-distributions-of-draws-from-the-approximate-posterior"></a>
</h2>
<p>The approximate posterior distributions can be viewed using the
<code><a href="../reference/plot_posterior_clocks_h.html">tickTack::plot_posterior_clocks_h</a></code> and
<code><a href="../reference/plot_posterior_weights_h.html">tickTack::plot_posterior_weights_h</a></code> functions, that
internally use functions from Bayesplot.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">posterior_clocks</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_posterior_clocks_h.html">plot_posterior_clocks_h</a></span><span class="op">(</span><span class="va">results</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">## Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">posterior_weights</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_posterior_weights_h.html">plot_posterior_weights_h</a></span><span class="op">(</span><span class="va">results</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="a4_Multiple_segment_timing_files/figure-html/unnamed-chunk-7-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="visualize-the-behavior-of-the-elbo-during-the-inference">Visualize the behavior of the ELBO during the inference<a class="anchor" aria-label="anchor" href="#visualize-the-behavior-of-the-elbo-during-the-inference"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">K</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">results_model_selection</span><span class="op">$</span><span class="va">model_selection_tibble</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p_elbo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">K</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">p_elbo</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_elbo_h.html">plot_elbo_h</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">elbo_iterations</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"K = "</span>, <span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">p_elbo</span> <span class="op">&lt;-</span> <span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">p_elbo</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>  <span class="co">#add global title</span></span></code></pre></div>
<p><img src="a4_Multiple_segment_timing_files/figure-html/unnamed-chunk-8-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_elbo</span></span></code></pre></div>
<pre><code><span><span class="co">## TableGrob (2 x 2) "arrange": 4 grobs</span></span>
<span><span class="co">##   z     cells    name           grob</span></span>
<span><span class="co">## 1 1 (1-1,1-1) arrange gtable[layout]</span></span>
<span><span class="co">## 2 2 (1-1,2-2) arrange gtable[layout]</span></span>
<span><span class="co">## 3 3 (2-2,1-1) arrange gtable[layout]</span></span>
<span><span class="co">## 4 4 (2-2,2-2) arrange gtable[layout]</span></span></code></pre>
<!-- ## Visualize model selection scores -->
<!-- ```{r} -->
<!-- #  -->
<!-- # p_model_selection <- tickTack::plot_model_selection_h(model_selection_tibble, best_K) -->
<!-- # p_model_selection -->
<!-- #  -->
<!-- # cat("Best K =",best_K) -->
<!-- ``` -->
</div>
<div class="section level2">
<h2 id="visualize-all-the-inference-results-for-each-k">Visualize all the inference results for each K<a class="anchor" aria-label="anchor" href="#visualize-all-the-inference-results-for-each-k"></a>
</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_model_selection_inference</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">K</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">plot_model_selection_inference</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_timing_h.html">plot_timing_h</a></span><span class="op">(</span><span class="va">results</span>, <span class="va">i</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"K = "</span>, <span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">plot_model_selection_inference</span> <span class="op">&lt;-</span> <span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">plot_model_selection_inference</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="co">#add global title</span></span></code></pre></div>
<p><img src="a4_Multiple_segment_timing_files/figure-html/unnamed-chunk-9-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_model_selection_inference</span></span></code></pre></div>
<pre><code><span><span class="co">## TableGrob (2 x 2) "arrange": 4 grobs</span></span>
<span><span class="co">##   z     cells    name           grob</span></span>
<span><span class="co">## 1 1 (1-1,1-1) arrange gtable[layout]</span></span>
<span><span class="co">## 2 2 (1-1,2-2) arrange gtable[layout]</span></span>
<span><span class="co">## 3 3 (2-2,1-1) arrange gtable[layout]</span></span>
<span><span class="co">## 4 4 (2-2,2-2) arrange gtable[layout]</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
