<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>4. Timing Clonal Peaks in a hierarchical fashioin • tickTack</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="4. Timing Clonal Peaks in a hierarchical fashioin">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tickTack</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/tickTack.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_introduction.html">1. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/a2_Single_segment_timing.html">2. Timing Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a3_Understanding_single_segment_timing.html">3. Understanding Timing of Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a4_Multiple_segment_timing.html">4. Timing Clonal Peaks in a hierarchical fashioin</a></li>
    <li><a class="dropdown-item" href="../articles/a5_Understanding_multiple_segments_timing.html">5. Understanding Timing of Multiple Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a6_Simulate_data.html">6. Simulate data</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>4. Timing Clonal Peaks in a hierarchical fashioin</h1>
            
      

      <div class="d-none name"><code>a4_Multiple_segment_timing.Rmd</code></div>
    </div>

    
    
<p>The <code>_fit_h</code> function in the <code>tickTack</code> package
estimates the timing of <code>K</code> clonal binomial peaks in cancer
genome sequencing data according to which a clustering can be performed.
This vignette describes the functionality of the <code>fit_h</code>
function, including input requirements, output, and an example analysis
using the <code><a href="../reference/pcawg_example_2.html">tickTack::pcawg_example_2</a></code> dataset.</p>
<div class="section level2">
<h2 id="overview-of-the-fit_h-function">Overview of the <code>fit_h</code> Function<a class="anchor" aria-label="anchor" href="#overview-of-the-fit_h-function"></a>
</h2>
<p>The <code>fit_h</code> function uses a hierarchical model to fit
clonal binomial peaks in sequencing data considering the grouping
structure of the chromosomes segments. It identifies segments of the
genome with specific karyotypes and mutations that meet the input
criteria, then estimates the timing of the groups of events and assign
each segment to a clock. The clock represents the pseudotime, where 0 is
the moment the cancerous population was born and 1 is the time of the
most recent common ancestor (MRCA) of the cells in the bulk sequencing
sample.</p>
<div class="section level3">
<h3 id="key-parameters">Key Parameters<a class="anchor" aria-label="anchor" href="#key-parameters"></a>
</h3>
<ul>
<li>
<strong><code>x</code></strong>: a CNAqc object with mutations, cna
and metadata</li>
<li>
<strong><code>max_attempts</code></strong>: Number of times the
variational inference is repeated to avoid local minima.</li>
<li>
<strong><code>INIT</code></strong>: Logical flag to pass some
initialization values tothe variational inference, default is
<code>TRURE</code>.</li>
<li>
<strong><code>tolerance</code></strong>: tolerance between two value
of subsequent iterations of gradient ascent on elbo, default is
<code>0.01</code>.</li>
<li>
<strong><code>possible_k</code></strong>: A character vector of
possible karyotypes, defaulting to
<code>c("2:1", "2:2", "2:0")</code>.</li>
<li>
<strong><code>alpha</code></strong>: Significance level, defaulting
to <code>0.05</code>.</li>
<li>
<strong><code>min_mutations_number</code></strong>: Minimum number
of mutations required for analysis, defaulting to <code>2</code>.</li>
<li>
<strong><code>n_components</code></strong>: If <code>0</code>, then
the strategy to choose the #components follows the default procedure,
otherwise the inference is repeated for K equal up to a maximun of
n_components.</li>
</ul>
</div>
<div class="section level3">
<h3 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h3>
<p>The function returns a list containing:</p>
<ol style="list-style-type: decimal">
<li>
<strong><code>data</code></strong>: The data used to perform the
inference after selecting the ones that respect the assumptions to be
used in the model.</li>
<li>
<strong><code>draws_and_summary</code></strong>: List of 3 for each
K the inference is performed with. Draws are available both for the
clocks and for the weights Summary statistics for the estimated timing
of clonal peaks.</li>
<li>
<strong><code>log_lik_matrix_list</code></strong>: Summary
statistics for the estimated timing of clonal peaks.</li>
<li>
<strong><code>elbo_iterations</code></strong>: Summary statistics
for the estimated timing of clonal peaks.</li>
</ol>
<p>If no segments meet the criteria, the function returns
<code>NULL</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="analyzing-ticktackpcawg_example_2">Analyzing <code>tickTack::pcawg_example_2</code><a class="anchor" aria-label="anchor" href="#analyzing-ticktackpcawg_example_2"></a>
</h2>
<p>We will use the <code><a href="../reference/pcawg_example_2.html">tickTack::pcawg_example_2</a></code> dataset to
demonstrate how to use the <code>fit_h</code> function.</p>
<div class="section level3">
<h3 id="input-data">Input Data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h3>
<p>The <code><a href="../reference/pcawg_example_2.html">tickTack::pcawg_example_2</a></code> dataset contains three
components:</p>
<ul>
<li>
<strong><code>mutations</code></strong>: Mutation data.</li>
<li>
<strong><code>cna</code></strong>: Copy number alterations
(CNA).</li>
<li>
<strong><code>metadata</code></strong>: Sample metadata, including
tumor purity.</li>
</ul>
<p>Preview the data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://caravagnalab.github.io/tickTack/">tickTack</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># View example dataset components</span></span>
<span><span class="va">mutations</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_example_2.html">pcawg_example_2</a></span><span class="op">$</span><span class="va">mutations</span></span>
<span><span class="va">cna</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_example_2.html">pcawg_example_2</a></span><span class="op">$</span><span class="va">cna</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="va"><a href="../reference/pcawg_example_2.html">pcawg_example_2</a></span><span class="op">$</span><span class="va">metadata</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mutations</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 45</span></span></span>
<span><span class="co">##   chr      from      to ref   alt      DP    NV   VAF sample      NR Hugo_Symbol</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> chr1  1<span style="text-decoration: underline;">018</span>754 1<span style="text-decoration: underline;">018</span>754 C     C        51    24 0.471 3b7810f…    27 C1orf159   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> chr1  1<span style="text-decoration: underline;">107</span>556 1<span style="text-decoration: underline;">107</span>556 C     C        53    17 0.321 3b7810f…    36 TTLL10     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> chr1  1<span style="text-decoration: underline;">127</span>192 1<span style="text-decoration: underline;">127</span>192 C     C        78    13 0.167 3b7810f…    65 TTLL10     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> chr1  1<span style="text-decoration: underline;">255</span>263 1<span style="text-decoration: underline;">255</span>263 C     C        66    23 0.348 3b7810f…    43 CPSF3L     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> chr1  1<span style="text-decoration: underline;">474</span>126 1<span style="text-decoration: underline;">474</span>126 C     C        68    17 0.25  3b7810f…    51 TMEM240    </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> chr1  1<span style="text-decoration: underline;">532</span>289 1<span style="text-decoration: underline;">532</span>289 G     G        39     6 0.154 3b7810f…    33 C1orf233   </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 34 more variables: Strand &lt;chr&gt;, Variant_Classification &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Variant_Type &lt;chr&gt;, Tumor_Seq_Allele2 &lt;chr&gt;, dbSNP_RS &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   dbSNP_Val_Status &lt;chr&gt;, Matched_Norm_Sample_Barcode &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Genome_Change &lt;chr&gt;, ref_context &lt;chr&gt;, gc_content &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   i_1000genomes_AF &lt;dbl&gt;, i_1000genomes_ID &lt;chr&gt;, i_Callers &lt;chr&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   i_GERM1000G &lt;lgl&gt;, i_GERMOVLP &lt;lgl&gt;, i_LOWSUPPORT &lt;lgl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   i_NORMALPANEL &lt;lgl&gt;, i_NumCallers &lt;dbl&gt;, i_OXOGFAIL &lt;lgl&gt;, …</span></span></span></code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cna</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 6 × 39</span></span></span>
<span><span class="co">##   chr        from        to Major minor   CCF total_cn  star level methods_agree</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> chr1      <span style="text-decoration: underline;">10</span>001 121<span style="text-decoration: underline;">499</span>999     1     1     1        2     3 a                 6</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">2</span> chr1  121<span style="text-decoration: underline;">500</span>000 128<span style="text-decoration: underline;">899</span>999     1     1     1        2     2 d                 4</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">3</span> chr1  128<span style="text-decoration: underline;">900</span>000 247<span style="text-decoration: underline;">247</span>500     1     1     1        2     3 a                 6</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">4</span> chr1  247<span style="text-decoration: underline;">247</span>501 249<span style="text-decoration: underline;">250</span>620     1     1     1        2     2 d                 4</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">5</span> chr2      <span style="text-decoration: underline;">66</span>017  90<span style="text-decoration: underline;">499</span>999     1     1     1        2     3 a                 6</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">6</span> chr2   90<span style="text-decoration: underline;">500</span>000  96<span style="text-decoration: underline;">799</span>999     1     1     1        2     3 a                 6</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 29 more variables: absolute_broad_major_cn &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   absolute_broad_minor_cn &lt;dbl&gt;, absolute_broad_het_error &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   absolute_broad_cov_error &lt;dbl&gt;, aceseq_copy_number &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   aceseq_minor_cn &lt;dbl&gt;, aceseq_major_cn &lt;dbl&gt;, battenberg_nMaj1_A &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   battenberg_nMin1_A &lt;dbl&gt;, battenberg_frac1_A &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   battenberg_nMaj2_A &lt;dbl&gt;, battenberg_nMin2_A &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   battenberg_frac2_A &lt;dbl&gt;, battenberg_SDfrac_A &lt;dbl&gt;, …</span></span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">metadata</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 1 × 6</span></span></span>
<span><span class="co">##   sample                  purity ploidy purity_conf_mad wgd_status wgd_uncertain</span></span>
<span><span class="co">##   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>        </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">1</span> 3b7810f7-f8ff-4d62-b76…  0.611   1.92           0.002 no_wgd     FALSE</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="running-the-fit_h-function">Running the <code>fit_h</code> function<a class="anchor" aria-label="anchor" href="#running-the-fit_h-function"></a>
</h3>
<p>We can run the <code>fit_h</code> function on the
<code><a href="../reference/pcawg_example_2.html">tickTack::pcawg_example_2</a></code> data to infer the timing of
clonal peaks</p>
</div>
<div class="section level3">
<h3 id="results">Results<a class="anchor" aria-label="anchor" href="#results"></a>
</h3>
<p>The <code>results</code> object that is returned together with the
CNAqc input object contains four components: <code>data</code>,
<code>draws_and_summary</code>, <code>log_lik_matrix_list</code> and
<code>elbo_iterations</code>. In the <code>data</code> there are both
the <code>input_data</code> passed to the Stan model and the
<code>accepted_cna</code> segments that respect the selection conditions
and are included in the inference. The <code>draws_and_summary</code>
part present an object for each inference performed with the used number
of components. Each of them is named by the number of components
<code>K</code> used for the inference so we can access them as
below.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View results</span></span>
<span><span class="va">fit</span><span class="op">$</span><span class="va">results_timing</span><span class="op">$</span><span class="va">data</span><span class="op">$</span><span class="va">accepted_cna</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 10 × 5</span></span></span>
<span><span class="co">##    segment_original_indx segment_name            segment_id karyotype chr  </span></span>
<span><span class="co">##                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>                    38 chr5_75225353_75780032           1 2:1       chr5 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>                    40 chr5_76857368_76998901           2 2:1       chr5 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>                    42 chr5_77822720_82754991           3 2:1       chr5 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>                    44 chr5_83468190_180689978          4 2:1       chr5 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>                    46 chr6_242501_4335355              5 2:0       chr6 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>                    48 chr6_4347801_29092265            6 2:0       chr6 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>                    50 chr6_29161839_38477499           7 2:0       chr6 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>                    54 chr7_65454_9937168               8 2:1       chr7 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>                    56 chr7_9946564_57999999            9 2:1       chr7 </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>                    58 chr7_61700000_158817500         10 2:1       chr7</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="va">fit</span><span class="op">$</span><span class="va">results_timing</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="interpreting-the-output">Interpreting the output<a class="anchor" aria-label="anchor" href="#interpreting-the-output"></a>
</h3>
<p>We can inspect the main output of interest to understand the timing
of clonal peaks. <code>results$draws_and_summary</code> contains: -
<strong><code>draws</code></strong> the draws from the approximate
posterior distribution of the taus and weights; -
<strong><code>summary</code></strong> a summary with the main statistics
of the approximate posterior distributions; -
<strong><code>summarized_results</code></strong> represents the clock
assignment, a tibble with the estimate of taus for each segment with a
copy number event that has been included in the hierarchical
inference</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View summary for a specific K, here K = 1</span></span>
<span><span class="va">inference_with_1_component</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">draws_and_summary</span><span class="op">[[</span><span class="st">"1"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">inference_with_5_component</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">$</span><span class="va">draws_and_summary</span><span class="op">[[</span><span class="st">"5"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># View detailed summarized results for a specific K, here K = 1 and K = 5</span></span>
<span><span class="va">inference_with_1_component</span><span class="op">$</span><span class="va">summarized_results</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 10 × 10</span></span></span>
<span><span class="co">##    segment_original_indx segment_name      segment_id karyotype chr   clock_mean</span></span>
<span><span class="co">##                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>                    38 chr5_75225353_75…          1 2:1       chr5       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>                    40 chr5_76857368_76…          2 2:1       chr5       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>                    42 chr5_77822720_82…          3 2:1       chr5       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>                    44 chr5_83468190_18…          4 2:1       chr5       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>                    46 chr6_242501_4335…          5 2:0       chr6       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>                    48 chr6_4347801_290…          6 2:0       chr6       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>                    50 chr6_29161839_38…          7 2:0       chr6       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>                    54 chr7_65454_99371…          8 2:1       chr7       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>                    56 chr7_9946564_579…          9 2:1       chr7       0.242</span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>                    58 chr7_61700000_15…         10 2:1       chr7       0.242</span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 4 more variables: clock_low &lt;dbl&gt;, clock_high &lt;dbl&gt;, alpha &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   beta &lt;dbl&gt;</span></span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">inference_with_5_component</span><span class="op">$</span><span class="va">summarized_results</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 10 × 10</span></span></span>
<span><span class="co">##    segment_original_indx segment_name      segment_id karyotype chr   clock_mean</span></span>
<span><span class="co">##                    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span>                    38 chr5_75225353_75…          1 2:1       chr5    0.368   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span>                    40 chr5_76857368_76…          2 2:1       chr5    0.000<span style="text-decoration: underline;">612</span></span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span>                    42 chr5_77822720_82…          3 2:1       chr5    0.026<span style="text-decoration: underline;">9</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span>                    44 chr5_83468190_18…          4 2:1       chr5    0.026<span style="text-decoration: underline;">9</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span>                    46 chr6_242501_4335…          5 2:0       chr6    0.955   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span>                    48 chr6_4347801_290…          6 2:0       chr6    0.955   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span>                    50 chr6_29161839_38…          7 2:0       chr6    0.955   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span>                    54 chr7_65454_99371…          8 2:1       chr7    0.167   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span>                    56 chr7_9946564_579…          9 2:1       chr7    0.167   </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span>                    58 chr7_61700000_15…         10 2:1       chr7    0.167   </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 4 more variables: clock_low &lt;dbl&gt;, clock_high &lt;dbl&gt;, alpha &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   beta &lt;dbl&gt;</span></span></span></code></pre>
</div>
<div class="section level3">
<h3 id="obtain-the-best-k-with-model_selection_h">Obtain the best K with model_selection_h<a class="anchor" aria-label="anchor" href="#obtain-the-best-k-with-model_selection_h"></a>
</h3>
<p>W e can run the <code>model_selection_h</code> function to obtain the
scores for each inference performed with a different K and take the one
with best ICL score if the BIC score prefer 2 components instead of 1,
otherwise choose 1 as best K. The function takes as input the
<code>results</code> and <code>n_components</code> and outputs the
<code>best_K</code> and the corresponding <code>best_fit</code> together
with the <code>model_selection_tibble</code> and the
<code>entropy_list</code> used to evaluate the ICL score.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results_model_selection</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/model_selection_h.html">model_selection_h</a></span><span class="op">(</span><span class="va">results</span>, n_components <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span>
<span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span>
<span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span>
<span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">best_K</span> <span class="op">&lt;-</span> <span class="va">results_model_selection</span><span class="op">$</span><span class="va">best_K</span></span>
<span><span class="va">model_selection_tibble</span> <span class="op">&lt;-</span> <span class="va">results_model_selection</span><span class="op">$</span><span class="va">model_selection_tibble</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visulizing-the-output">Visulizing the output<a class="anchor" aria-label="anchor" href="#visulizing-the-output"></a>
</h2>
<p>The results can be viewed is genome-wise perspective using the
<code><a href="../reference/plot_timing_h.html">tickTack::plot_timing_h</a></code> function.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># fit$metadata = tibble(purity=fit$purity)</span></span>
<span><span class="co"># fit$cna &lt;- fit$cna%&gt;%mutate(Major_2=Major,minor_2=minor)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'dplyr'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     filter, lag</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     intersect, setdiff, setequal, union</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fit</span><span class="op">$</span><span class="va">reference_genome</span> <span class="op">&lt;-</span> <span class="st">"GRCh37"</span></span>
<span><span class="va">fit</span><span class="op">$</span><span class="va">cna</span> <span class="op">&lt;-</span> <span class="va">fit</span><span class="op">$</span><span class="va">cna</span><span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Major_2<span class="op">=</span><span class="va">Major</span>,minor_2<span class="op">=</span><span class="va">minor</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/plot_cnaqc.html">plot_cnaqc</a></span><span class="op">(</span><span class="va">fit</span>,add_mobster<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span>
<span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span>
<span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span>
<span><span class="co">## Warning: Some Pareto k diagnostic values are too high. See help('pareto-k-diagnostic') for details.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">CNAqc</span>, <span style="font-style: italic;">'Copy Number Alteration quality check'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/CNAqc/&gt;</span></span></span></code></pre>
<pre><code><span><span class="co">## </span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">CNAqc - CNA Quality Check</span> <span style="color: #00BBBB;">───────────────────────────────────────────────────</span></span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Using reference genome coordinates for: <span style="color: #00BB00;">GRCh37</span>.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #BBBB00;">!</span> Detected indels mutation (substitutions with &gt;1 reference/alternative nucleotides).</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #BB0000;">✖</span> NA values in some of the required mutation columns, these will be removed.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BB00;">✔</span> Fortified calls for <span style="color: #00BB00;">41400</span> somatic mutations: <span style="color: #00BB00;">36223</span> SNVs (<span style="color: #00BB00;">87%</span>) and <span style="color: #00BB00;">5177</span> indels.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BBBB;">ℹ</span> <span style="color: #00BB00;">2</span> subclonal CNAs detected in the data.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #BBBB00;">!</span> Added segments length (in basepairs) to CNA segments.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BB00;">✔</span> Fortified CNAs for <span style="color: #00BB00;">107</span> segments: <span style="color: #00BB00;">105</span> clonal and <span style="color: #00BB00;">2</span> subclonal.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">39324</span> mutations mapped to clonal CNAs.</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Unknown or uninitialised column: `mutations`.</span></span></code></pre>
<pre><code><span><span class="co">## <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">0</span> mutations mapped to subclonal CNAs.</span></span></code></pre>
<pre><code><span><span class="co">## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span></span>
<span><span class="co">## <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">CNAqc</span> package.</span></span>
<span><span class="co">##   Please report the issue at <span style="color: #0000BB; font-style: italic;">&lt;https://github.com/caravagnalab/CNAqc/issues&gt;</span>.</span></span>
<span><span class="co">## <span style="color: #555555;">This warning is displayed once every 8 hours.</span></span></span>
<span><span class="co">## <span style="color: #555555;">Call `lifecycle::last_lifecycle_warnings()` to see where this warning was</span></span></span>
<span><span class="co">## <span style="color: #555555;">generated.</span></span></span></code></pre>
<pre><code><span><span class="co">## Warning: Removed 17 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">## (`geom_segment()`).</span></span></code></pre>
<p><img src="a4_Multiple_segment_timing_files/figure-html/unnamed-chunk-6-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="visualize-distributions-of-draws-from-the-approximate-posterior">Visualize distributions of draws from the approximate posterior<a class="anchor" aria-label="anchor" href="#visualize-distributions-of-draws-from-the-approximate-posterior"></a>
</h2>
<p>The approximate posterior distributions can be viewed using the
<code><a href="../reference/plot_posterior_clocks_h.html">tickTack::plot_posterior_clocks_h</a></code> and
<code><a href="../reference/plot_posterior_weights_h.html">tickTack::plot_posterior_weights_h</a></code> functions, that
internally use functions from Bayesplot.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">posterior_clocks</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_posterior_clocks_h.html">plot_posterior_clocks_h</a></span><span class="op">(</span><span class="va">results</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Scale for <span style="color: #00BB00;">x</span> is already present.</span></span>
<span><span class="co">## Adding another scale for <span style="color: #00BB00;">x</span>, which will replace the existing scale.</span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">posterior_weights</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_posterior_weights_h.html">plot_posterior_weights_h</a></span><span class="op">(</span><span class="va">results</span>, <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="a4_Multiple_segment_timing_files/figure-html/unnamed-chunk-7-1.png" width="576" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="visualize-the-behavior-of-the-elbo-during-the-inference">Visualize the behavior of the ELBO during the inference<a class="anchor" aria-label="anchor" href="#visualize-the-behavior-of-the-elbo-during-the-inference"></a>
</h2>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">K</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">results_model_selection</span><span class="op">$</span><span class="va">model_selection_tibble</span><span class="op">)</span></span>
<span></span>
<span><span class="va">p_elbo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">K</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">p_elbo</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_elbo_h.html">plot_elbo_h</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">elbo_iterations</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"K = "</span>, <span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">p_elbo</span> <span class="op">&lt;-</span> <span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">p_elbo</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span>  <span class="co">#add global title</span></span></code></pre></div>
<p><img src="a4_Multiple_segment_timing_files/figure-html/unnamed-chunk-8-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_elbo</span></span></code></pre></div>
<pre><code><span><span class="co">## TableGrob (3 x 2) "arrange": 5 grobs</span></span>
<span><span class="co">##   z     cells    name           grob</span></span>
<span><span class="co">## 1 1 (1-1,1-1) arrange gtable[layout]</span></span>
<span><span class="co">## 2 2 (1-1,2-2) arrange gtable[layout]</span></span>
<span><span class="co">## 3 3 (2-2,1-1) arrange gtable[layout]</span></span>
<span><span class="co">## 4 4 (2-2,2-2) arrange gtable[layout]</span></span>
<span><span class="co">## 5 5 (3-3,1-1) arrange gtable[layout]</span></span></code></pre>
<!-- ## Visualize model selection scores -->
<!-- ```{r} -->
<!-- #  -->
<!-- # p_model_selection <- tickTack::plot_model_selection_h(model_selection_tibble, best_K) -->
<!-- # p_model_selection -->
<!-- #  -->
<!-- # cat("Best K =",best_K) -->
<!-- ``` -->
</div>
<div class="section level2">
<h2 id="visualize-all-the-inference-results-for-each-k">Visualize all the inference results for each K<a class="anchor" aria-label="anchor" href="#visualize-all-the-inference-results-for-each-k"></a>
</h2>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_model_selection_inference</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">K</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="va">plot_model_selection_inference</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">tickTack</span><span class="fu">::</span><span class="fu"><a href="../reference/plot_timing_h.html">plot_timing_h</a></span><span class="op">(</span><span class="va">results</span>, <span class="va">i</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"K = "</span>, <span class="va">i</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">plot_model_selection_inference</span> <span class="op">&lt;-</span> <span class="fu">gridExtra</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gridExtra/man/arrangeGrob.html" class="external-link">grid.arrange</a></span><span class="op">(</span>grobs <span class="op">=</span> <span class="va">plot_model_selection_inference</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="co">#add global title</span></span></code></pre></div>
<p><img src="a4_Multiple_segment_timing_files/figure-html/unnamed-chunk-9-1.png" width="576" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plot_model_selection_inference</span></span></code></pre></div>
<pre><code><span><span class="co">## TableGrob (3 x 2) "arrange": 5 grobs</span></span>
<span><span class="co">##   z     cells    name           grob</span></span>
<span><span class="co">## 1 1 (1-1,1-1) arrange gtable[layout]</span></span>
<span><span class="co">## 2 2 (1-1,2-2) arrange gtable[layout]</span></span>
<span><span class="co">## 3 3 (2-2,1-1) arrange gtable[layout]</span></span>
<span><span class="co">## 4 4 (2-2,2-2) arrange gtable[layout]</span></span>
<span><span class="co">## 5 5 (3-3,1-1) arrange gtable[layout]</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
