<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>5. Understanding Timing of Multiple Clonal Peaks • tickTack</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="5. Understanding Timing of Multiple Clonal Peaks">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tickTack</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/tickTack.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_introduction.html">1. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/a2_Single_segment_timing.html">2. Timing Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a3_Understanding_single_segment_timing.html">3. Understanding Timing of Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a4_Multiple_segment_timing.html">4. Timing Clonal Peaks in a hierarchical fashioin</a></li>
    <li><a class="dropdown-item" href="../articles/a5_Understanding_multiple_segments_timing.html">5. Understanding Timing of Multiple Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a6_Simulate_data.html">6. Simulate data</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>5. Understanding Timing of Multiple Clonal Peaks</h1>
            
      

      <div class="d-none name"><code>a5_Understanding_multiple_segments_timing.Rmd</code></div>
    </div>

    
    
<p>This vignette delves into how the <code>tickTack</code> package
obtains timing results for clonal peaks using the <code>fit_h</code>
function. It explains the underlying processes, including the
computation of clonal peaks and the Stan model used for inference.</p>
<div class="section level2">
<h2 id="key-components-of-the-analysis">Key Components of the Analysis<a class="anchor" aria-label="anchor" href="#key-components-of-the-analysis"></a>
</h2>
<p>This function performs hierarchical Bayesian inference using ADVI to
approximate the posterior distributions of clock parameters for copy
number alterations (CNAs). It iteratively fits models with different
numbers of components (up to <code>k_max</code>), computes posterior
summaries, and logs ELBO (Evidence Lower Bound) values for each fit.</p>
<p>The input data are: - x list: A <code>CNAqc</code> object containing
mutation and CNA data. - max_attempts num: Maximum number of repeated
inference attempts for ADVI. - INIT logical: Whether to perform
initialization of parameters. - tolerance num: Tolerance level for ELBO
optimization. - possible_k chr: Possible karyotype configurations (e.g.,
<code>"2:1"</code>, <code>"2:2"</code>, <code>"2:0"</code>). - alpha
num: Confidence interval level for binomial filtering. -
min_mutations_number num: Minimum number of mutations required for a
segment to be included in inference. - n_components num: Number of
components specified by the user (overrides automatic
determination).</p>
<p>It returns a list containing processed input data, posterior draws,
ELBO iterations, and other inference results.</p>
</div>
<div class="section level2">
<h2 id="steps-of-the-fit_h-function">Steps of the <code>fit_h</code> Function<a class="anchor" aria-label="anchor" href="#steps-of-the-fit_h-function"></a>
</h2>
<div class="section level3">
<h3 id="step-1-validate-input-data">Step 1: Validate Input Data<a class="anchor" aria-label="anchor" href="#step-1-validate-input-data"></a>
</h3>
<ul>
<li>Extract mutation and CNA data from the <code>CNAqc</code>
object.</li>
<li>Ensure both mutations and CNA data are non-empty. If empty, the
function terminates with an error.</li>
</ul>
</div>
<div class="section level3">
<h3 id="step-2-prepare-input-data">Step 2: Prepare Input Data<a class="anchor" aria-label="anchor" href="#step-2-prepare-input-data"></a>
</h3>
<ul>
<li>Use <code>prepare_input_data</code> to filter and format mutation
and CNA data based on the provided karyotypes (<code>possible_k</code>),
confidence intervals (<code>alpha</code>), and minimum mutations per
segment (<code>min_mutations_number</code>).</li>
<li>Extract <code>accepted_cna</code> and <code>input_data</code> for
inference.</li>
</ul>
</div>
<div class="section level3">
<h3 id="step-3-determine-maximum-number-of-components-k_max">Step 3: Determine Maximum Number of Components
(<code>k_max</code>)<a class="anchor" aria-label="anchor" href="#step-3-determine-maximum-number-of-components-k_max"></a>
</h3>
<ul>
<li>If <code>n_components</code> is not provided, calculate
<code>k_max</code> based on the number of segments (<code>S</code>)
using predefined rules.</li>
</ul>
</div>
<div class="section level3">
<h3 id="step-4-iterative-inference-for-each-component-count-k">Step 4: Iterative Inference for Each Component Count
(<code>K</code>)<a class="anchor" aria-label="anchor" href="#step-4-iterative-inference-for-each-component-count-k"></a>
</h3>
<p>For each <code>K</code> from 1 to <code>k_max</code>: - Add
<code>K</code> to the <code>input_data</code>. - If <code>INIT</code> is
enabled, initialize parameters using <code>get_initialization</code>. -
Fit the model using <code>fit_variational_h</code> with ADVI. - Extract
relevant posterior variables (<code>tau</code> and <code>w</code>) and
compute summaries. - Compute clock assignments for segments using
<code>compute_clock_assignment</code>. - Store results, including
posterior summaries and clock assignments.</p>
<p>The model to fit is described by the following probabilistic
graphical model.
<img src="img/pgm.png" alt="Overview timing problem" width="650" style="display: block; margin: 0 auto"><br></p>
<p>The model estimates the posterior distribution of <code>tau</code>
and <code>weights</code> , which represents the timing of clonal
expansions.</p>
<p>The stan code used for the model is divided in the following
pieces.</p>
<div class="section level4">
<h4 id="data-block">Data block<a class="anchor" aria-label="anchor" href="#data-block"></a>
</h4>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>data {</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  <span class="sc">/</span><span class="er">/</span> input the karyotyoe to specify the formula <span class="cf">for</span> theta</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>  int S;                        <span class="sc">/</span><span class="er">/</span> number of segments</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  int K;                        <span class="sc">/</span><span class="er">/</span> clocks</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  int N;                        <span class="sc">/</span><span class="er">/</span> total number of mutations</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  array[S] int karyotype;       <span class="sc">/</span><span class="er">/</span> list of karyotype associated wit the segments</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>  array[N] int seg_assignment;  <span class="sc">/</span><span class="er">/</span> segment_id assignment to the mutations</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a>  array[S,<span class="dv">2</span>] real peaks;        <span class="sc">/</span><span class="er">/</span><span class="cf">for</span> each segment, S vectors of dim <span class="dv">2</span></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a>  array[N] int NV;              <span class="sc">/</span><span class="er">/</span> <span class="cf">for</span> all the segments</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>  array[N] int DP;</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="parameters-block">Parameters block<a class="anchor" aria-label="anchor" href="#parameters-block"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>parameters {</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  array[S] simplex[K] w;              <span class="sc">/</span><span class="er">/</span> simplex[K] w[N] mixing proportions <span class="cf">for</span> each segment group; check prior should be more on the higher values</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  vector<span class="sc">&lt;</span>lower<span class="ot">=</span><span class="fl">0.0001</span>,upper<span class="ot">=</span><span class="fl">0.999</span><span class="sc">&gt;</span>[K] tau;     <span class="sc">/</span><span class="er">/</span>clocks   ordered[K] tau  vector<span class="sc">&lt;</span>lower<span class="ot">=</span><span class="dv">0</span>,upper<span class="ot">=</span><span class="dv">1</span><span class="sc">&gt;</span>[K]</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="sc">/</span><span class="er">/</span>vector[K] alpha;</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  simplex[K] phi;</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  real<span class="sc">&lt;</span>lower<span class="ot">=</span><span class="dv">0</span><span class="sc">&gt;</span> kappa;</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="transformed-parameters-block">Transformed parameters block<a class="anchor" aria-label="anchor" href="#transformed-parameters-block"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>transformed parameters{</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  array[S] vector[K] perturbed_w;  <span class="sc">/</span><span class="er">/</span> Pesi perturbati e rinormalizzati</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  real<span class="sc">&lt;</span>lower<span class="ot">=</span><span class="dv">0</span><span class="sc">&gt;</span> epsilon;</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  epsilon <span class="ot">=</span> <span class="fl">0.01</span>;</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="cf">for</span> (s <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>S) {</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>    <span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>K) {</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>      <span class="cf">if</span> (w[s][k] <span class="sc">&gt;</span> <span class="fl">0.5</span>) {</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>        perturbed_w[s][k] <span class="ot">=</span> w[s][k] <span class="sc">-</span> epsilon;</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>        perturbed_w[s][k] <span class="ot">=</span> w[s][k] <span class="sc">+</span> epsilon;</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>      }</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>    }</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>    <span class="sc">/</span><span class="er">/</span> Rinormalizza i pesi</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>    perturbed_w[s] <span class="ot">=</span> perturbed_w[s] <span class="sc">/</span> <span class="fu">sum</span>(perturbed_w[s]);</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>  }</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>  array[S,K,<span class="dv">2</span>] real<span class="sc">&lt;</span>lower<span class="ot">=</span><span class="dv">0</span>,upper<span class="ot">=</span><span class="dv">1</span><span class="sc">&gt;</span> theta; <span class="sc">/</span><span class="er">/</span>binomial mixing proportions <span class="sc">/</span><span class="er">/</span> array[S,K] simplex[<span class="dv">2</span>] theta;</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>  <span class="cf">for</span> (s <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>S){</span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>   <span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>K){</span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>      <span class="cf">if</span> (karyotype[s] <span class="sc">==</span> <span class="dv">1</span>) {</span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>        theta[s,k,<span class="dv">1</span>] <span class="ot">=</span> (<span class="dv">3</span> <span class="sc">-</span> <span class="dv">2</span><span class="sc">*</span>tau[k]) <span class="sc">/</span> (<span class="dv">3</span> <span class="sc">-</span> tau[k]);      <span class="sc">/</span><span class="er">/</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">1</span></span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a>        theta[s,k,<span class="dv">2</span>] <span class="ot">=</span> tau[k] <span class="sc">/</span> (<span class="dv">3</span> <span class="sc">-</span> tau[k]);</span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a>      }</span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a>      <span class="cf">else</span> {</span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a>        theta[s,k,<span class="dv">1</span>] <span class="ot">=</span> (<span class="dv">2</span> <span class="sc">-</span> <span class="dv">2</span><span class="sc">*</span>tau[k]) <span class="sc">/</span> (<span class="dv">2</span> <span class="sc">-</span> tau[k]);      <span class="sc">/</span><span class="er">/</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">0</span> <span class="sc">-</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">2</span></span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a>        theta[s,k,<span class="dv">2</span>] <span class="ot">=</span> tau[k]<span class="sc">/</span>(<span class="dv">2</span> <span class="sc">-</span> tau[k]);</span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a>      }</span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a>    }</span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a>  }</span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a></span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a>  vector[K] alpha <span class="ot">=</span> kappa <span class="sc">*</span> phi;</span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="section level4">
<h4 id="model-block">Model block<a class="anchor" aria-label="anchor" href="#model-block"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>model {</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  vector[K<span class="sc">*</span><span class="dv">2</span>] contributions;              <span class="sc">/</span><span class="er">/</span>real contributions[K <span class="sc">*</span> <span class="dv">2</span>];  <span class="sc">/</span><span class="er">/</span> Array to hold contributions <span class="cf">for</span> log_sum_exp</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="sc">/</span><span class="er">/</span> priors</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  phi <span class="sc">~</span> <span class="fu">dirichlet</span>(<span class="fu">rep_vector</span>(<span class="fl">1.0</span>, K));;</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  kappa <span class="sc">~</span> <span class="fu">gamma</span>(<span class="dv">2</span>, <span class="fl">0.5</span>);                  <span class="sc">/</span><span class="er">/</span> strictly positive with a long right tail.</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>                                          <span class="sc">/</span><span class="er">/</span> phi <span class="ot">=</span> expected value of w, <span class="fu">kappa</span> (minus K) <span class="ot">=</span> concentrazione della distribuzione <span class="sc">/</span> strength of the prior mean measured <span class="cf">in</span> number of prior observations.</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>  <span class="cf">for</span> (s <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>S){</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>      w[s] <span class="sc">~</span> <span class="fu">dirichlet</span>(alpha);</span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  }</span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>  <span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>K) {</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>    tau[k] <span class="sc">~</span> <span class="fu">beta</span>(<span class="dv">2</span>,<span class="dv">2</span>);                   <span class="sc">/</span><span class="er">/</span> Beta prior <span class="cf">for</span> tau</span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>  }</span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>  </span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>  <span class="sc">/</span><span class="er">/</span>likelihood</span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N) {</span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>    int c <span class="ot">=</span> <span class="dv">1</span>;</span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a>    </span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>    <span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>K) {</span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a>      <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) {</span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a>        </span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>        real contribution <span class="ot">=</span> <span class="fu">log</span>(perturbed_w[seg_assignment[i], k]) <span class="sc">+</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a>                            <span class="fu">log</span>(theta[seg_assignment[i], k, j]) <span class="sc">+</span></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>                            <span class="fu">binomial_lpmf</span>(NV[i] <span class="sc">|</span> DP[i], peaks[seg_assignment[i], j]);</span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>        contributions[c] <span class="ot">=</span> contribution;</span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>        c <span class="sc">+</span><span class="er">=</span> <span class="dv">1</span>;</span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>      }</span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a>    }</span>
<span id="cb4-33"><a href="#cb4-33" tabindex="-1"></a>    target <span class="sc">+</span><span class="er">=</span> <span class="fu">log_sum_exp</span>(contributions);</span>
<span id="cb4-34"><a href="#cb4-34" tabindex="-1"></a>  }</span>
<span id="cb4-35"><a href="#cb4-35" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" tabindex="-1"></a>}</span></code></pre></div>
<p>The <code>model</code> block contains three components:</p>
<ul>
<li><p><strong>Prior</strong>: The line <code>tau[k] ~ beta(2,2)</code>
applies a Beta prior to the components’ values. The line
<code>w[s] ~ dirichlet(alpha);</code> applies a Dirichlet prior to the
mixture probabilities that depends on alpha = phi * kappa. This means
that we do not a priori favor one clock component over the other unless
there is data evidence. The lines
<code>phi ~ dirichlet(rep_vector(1.0, K));</code> and
<code>kappa ~ gamma(2, 0.5);</code> apply the priors to the
reparametrization values for the alpha parameter of the
weights.</p></li>
<li><p><strong>Likelihood</strong>: The likelihood is computed using the
<code>binomial_lpmf(NV[i] | DP[i], peaks[seg_assignment[i], j])</code>
(or the Negative binomial). This term computes the probability of
observing <code>NV[i]</code> variant reads given sequencing depth
<code>DP[i]</code> and a given theoretical clonal peak value
<code>peaks[k]</code>.</p></li>
<li><p><strong>Marginalization</strong>: The line
<code>log_sum_exp(contributions)</code> computes the log marginal
likelihood by summing over possible contributions (karyotypes) using the
softmax approach, allowing the model to integrate over possible
karyotype contributions (<code>omega</code>).</p></li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="generated-quantities">Generated quantities<a class="anchor" aria-label="anchor" href="#generated-quantities"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>generated quantities {</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="sc">/</span><span class="er">/</span>array[N] vector[K<span class="sc">*</span><span class="dv">2</span>] log_lik_matrix;</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  </span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>  vector[N] log_lik;                        <span class="sc">/</span><span class="er">/</span> log<span class="sc">-</span>likelihood <span class="cf">for</span> each data point</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>N) {</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>    vector[K<span class="sc">*</span><span class="dv">2</span>] contributions;</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>    int c <span class="ot">=</span> <span class="dv">1</span>;</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>    <span class="cf">for</span> (k <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>K) {</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>      <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) {</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>        contributions[c] <span class="ot">=</span> <span class="fu">log</span>(w[seg_assignment[i],k]) <span class="sc">+</span> <span class="fu">log</span>(theta[seg_assignment[i],k,j]) <span class="sc">+</span> <span class="fu">binomial_lpmf</span>(NV[i] <span class="sc">|</span> DP[i], peaks[seg_assignment[i],j]);</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>        c <span class="sc">+</span><span class="er">=</span> <span class="dv">1</span>;</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>      }</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>    }</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>    log_lik[i] <span class="ot">=</span> <span class="fu">log_sum_exp</span>(contributions);</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>    <span class="sc">/</span><span class="er">/</span>log_lik_matrix[i] <span class="ot">=</span> contributions;</span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a>  }</span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-5-log-elbo-iterations">Step 5: Log ELBO Iterations<a class="anchor" aria-label="anchor" href="#step-5-log-elbo-iterations"></a>
</h3>
<ul>
<li>Extract ELBO values from the output files of the model fit and save
them for each <code>K</code>.</li>
</ul>
</div>
<div class="section level3">
<h3 id="step-6-compile-results">Step 6: Compile Results<a class="anchor" aria-label="anchor" href="#step-6-compile-results"></a>
</h3>
<ul>
<li>Store posterior draws, summaries, log-likelihood contributions, and
ELBO iterations in lists for each <code>K</code>.</li>
<li>Return the compiled results as a list:
<ul>
<li>
<code>data</code>: Processed input data.</li>
<li>
<code>draws_and_summary</code>: Posterior draws and summaries for
each <code>K</code>.</li>
<li>
<code>log_lik_matrix_list</code>: Log-likelihood matrices for each
<code>K</code>.</li>
<li>
<code>elbo_iterations</code>: ELBO iteration data for each
<code>K</code>.</li>
</ul>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="step-7-update-cnaqc-object">Step 7: Update CNAqc Object<a class="anchor" aria-label="anchor" href="#step-7-update-cnaqc-object"></a>
</h3>
<ul>
<li>Add the results to the <code>CNAqc</code> object as
<code>results_timing</code>.</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
