<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>tickTack • tickTack</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="tickTack">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">tickTack</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/tickTack.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/a1_introduction.html">1. Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/a2_Single_segment_timing.html">2. Timing Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a3_Understanding_single_segment_timing.html">3. Understanding Timing of Clonal Peaks</a></li>
    <li><a class="dropdown-item" href="../articles/a4_Multiple_segment_timing.html">4. Timing Clonal Peaks in a hierarchical fashioin</a></li>
    <li><a class="dropdown-item" href="../articles/a5_Understanding_multiple_segments_timing.html">5. Understanding Timing of Multiple Clonal Peaks</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>tickTack</h1>
            
      

      <div class="d-none name"><code>tickTack.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><code>tickTack</code> requires as input a <code>CNAqc</code> object
with attributes <code>cna</code>, <code>mutations</code> and
<code>metadata</code>. The main input for the tool are:</p>
<ul>
<li>the read counts from somatic mutations such as single-nucleotide
variants (SNVs) in the mutation attribute;</li>
<li>allele-specific copy number segments (CNAs) for clonal segments must
be encoded in the cna attribute;</li>
<li>a tumor purity estimate in the metadata.</li>
</ul>
<p>The tool uses chromosome coordinates to map mutations to segments.
The conversion of relative to absolute genome coordinates requires to
fix a reference genome build; supported reference is GRCh38/hg17 that is
also supported in CNAqc.</p>
<p><code>tickTack</code> can be used to:</p>
<ul>
<li>time the genomic segments affected by a Copy Number event, obtaining
one clock per segment</li>
<li>time multiple CNAs in a hierarchical fashion, identifying
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>K</mi><annotation encoding="application/x-tex">K</annotation></semantics></math>
clocks that cluster some segemnts together.</li>
</ul>
<p>The following concepts are used to infer copy number timing.</p>
<div class="section level3">
<h3 id="vaf-peaks">VAF peaks<a class="anchor" aria-label="anchor" href="#vaf-peaks"></a>
</h3>
<p>The point mutations that are present on the duplicated region are
duplicated in copy with the segment. Therefore we can use the proportion
of mutations happede before and after the Copy Number event
distinguishing between mutations in single copy and double copies.</p>
<p><img src="img%2Ftiming_problem.png" alt="Overview timing problem" width="650" style="display: block; margin: 0 auto"><br></p>
<p>Therefore, for a single segment the value of the clock associated
with the Copy Number event is obtained as a transformation from the
proportions of mutations in single and double copy. The following
quantities need to be considered:</p>
<p><img src="img%2FMutations_single_double_copy_draw_process.png" alt="Overview timing problem" width="650" style="display: block; margin: 0 auto"><br></p>
<p>In the case of a trisomy without LOH, we can consider the fact that,
before
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>,
1 chromosome will accumulate mutations that will duplicate, while the
other will accumulate mutations that will remain in single copy. On the
other hand, after
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>,
both chromosomes will accumulate mutations which will remain in single
copy. Therefore one can write the system:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtable><mtr><mtd columnalign="right" style="text-align: right"><mrow><mo stretchy="true" form="prefix">{</mo><mtable><mtr><mtd columnalign="left" style="text-align: left"><msub><mi>N</mi><mn>2</mn></msub><mo>=</mo><mi>ρ</mi><mi>τ</mi><msub><mi>N</mi><mn>1</mn></msub><mo>=</mo><mi>ρ</mi><mi>τ</mi><mo>+</mo><mn>3</mn><mi>ρ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>τ</mi><mo stretchy="true" form="postfix">)</mo></mrow></mtd></mtr></mtable></mrow></mtd></mtr></mtable><annotation encoding="application/x-tex">\begin{align}
    \begin{cases}
        N_2 = \rho \tau  
        N_1 = \rho \tau + 3\rho(1 - \tau) \nonumber 
    \end{cases}
\end{align}</annotation></semantics></math></p>
<p>Using the first one to obtain
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>ρ</mi><annotation encoding="application/x-tex">\rho</annotation></semantics></math>
and inserting into the second one, the solution for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
becomes:</p>
<p><span class="math display">$$\begin{equation}
    N_1 = N_2 + \frac{3N_2}{\tau}(1-\tau) \hspace{2mm} \rightarrow
\hspace{2mm} N_1 + 2N_2 = \frac{3N_2}{\tau} \hspace{2mm} \rightarrow
\hspace{2mm} \tau = \frac{3N_2}{N_1 + 2N_2}
\end{equation}$$</span></p>
<p>The case of the CNLOH and of the segment doubling can be treated
together. In fact, in the first case, before
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
the mutations that will duplicate accumulate on a single chromosome and
after
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
the mutations that will remain in a single copy accumulate on two
chromosomes. The system therefore becomes:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtable><mtr><mtd columnalign="right" style="text-align: right"><mrow><mo stretchy="true" form="prefix">{</mo><mtable><mtr><mtd columnalign="left" style="text-align: left"><msub><mi>N</mi><mn>2</mn></msub><mo>=</mo><mi>ρ</mi><mi>τ</mi><msub><mi>N</mi><mn>1</mn></msub><mo>=</mo><mn>2</mn><mi>ρ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>τ</mi><mo stretchy="true" form="postfix">)</mo></mrow></mtd></mtr></mtable></mrow></mtd></mtr></mtable><annotation encoding="application/x-tex">\begin{align}
    \begin{cases}
        N_2 = \rho \tau  
        N_1 = 2\rho(1 - \tau) \nonumber 
    \end{cases}
\end{align}</annotation></semantics></math></p>
<p>A very similar things happens in the case of the 2:2, with the only
difference that the number of chromosomes accumulating a certain type of
mutation will be double, both after and before
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>.
Hence, the system becomes:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mtable><mtr><mtd columnalign="right" style="text-align: right"><mrow><mo stretchy="true" form="prefix">{</mo><mtable><mtr><mtd columnalign="left" style="text-align: left"><msub><mi>N</mi><mn>2</mn></msub><mo>=</mo><mn>2</mn><mi>ρ</mi><mi>τ</mi><msub><mi>N</mi><mn>1</mn></msub><mo>=</mo><mn>4</mn><mi>ρ</mi><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>τ</mi><mo stretchy="true" form="postfix">)</mo></mrow></mtd></mtr></mtable></mrow></mtd></mtr></mtable><annotation encoding="application/x-tex">\begin{align}
    \begin{cases}
        N_2 = 2\rho \tau  
        N_1 = 4\rho(1 - \tau) \nonumber 
    \end{cases}
\end{align}</annotation></semantics></math></p>
<p>Therefore, the two system can be solved similarly (you can simply
drop a factor of 2 in the second case). The solution for
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>τ</mi><annotation encoding="application/x-tex">\tau</annotation></semantics></math>
easily becomes:</p>
<p><span class="math display">$$\begin{equation}
    N_1 = \frac{2N_2(1-\tau)}{\tau} \hspace{2mm} \rightarrow
\hspace{2mm} \tau(N_1 + 2N_2) = 2N_2 \hspace{2mm} \rightarrow
\hspace{2mm} \tau = \frac{2N_2}{2N_2 + N_1}
\end{equation}$$</span></p>
</div>
<div class="section level3">
<h3 id="clonal-cnas">Clonal CNAs<a class="anchor" aria-label="anchor" href="#clonal-cnas"></a>
</h3>
<p>Consider:</p>
<ul>
<li>mutations sitting on a segment
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mi>A</mi><mo>:</mo><mi>n</mi><mi>B</mi></mrow><annotation encoding="application/x-tex">nA:nB</annotation></semantics></math>;</li>
<li>tumour purity
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>π</mi><annotation encoding="application/x-tex">\pi</annotation></semantics></math>;</li>
<li>a healthy diploid normal;</li>
</ul>
<p>Since the proportion of all reads from the tumour is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>π</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>+</mo><msub><mi>n</mi><mi>B</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">\pi(n_A+n_B)</annotation></semantics></math>,
and from the normal is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>π</mi><mo stretchy="true" form="postfix">)</mo></mrow></mrow><annotation encoding="application/x-tex">2(1-\pi)</annotation></semantics></math>.
Then, muations present in
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>m</mi><annotation encoding="application/x-tex">m</annotation></semantics></math>
copies of the tumour genome should peak at VAF value</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>v</mi><mi>m</mi></msub><mrow><mo stretchy="true" form="prefix">(</mo><mi>c</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>=</mo><mfrac displaystyle="true"><mrow><mi>m</mi><mi>π</mi><mi>c</mi></mrow><mrow><mn>2</mn><mrow><mo stretchy="true" form="prefix">(</mo><mn>1</mn><mo>−</mo><mi>π</mi><mo stretchy="true" form="postfix">)</mo></mrow><mo>+</mo><mi>π</mi><mrow><mo stretchy="true" form="prefix">(</mo><msub><mi>n</mi><mi>A</mi></msub><mo>+</mo><msub><mi>n</mi><mi>B</mi></msub><mo stretchy="true" form="postfix">)</mo></mrow></mrow></mfrac><mspace width="0.167em"></mspace><mi>.</mi></mrow><annotation encoding="application/x-tex">
v_m(c) = \dfrac{m \pi c}{
2 (1 - \pi) + \pi (n_A+n_B)
} \, .
</annotation></semantics></math></p>
<!-- ## The model -->
<!-- <img src="/Users/jovoni/GitHub/tickTack/vignettes/img/pgm.png" alt="Overview timing problem" width="650" style="display: block; margin: 0 auto"/><br> -->
<!-- Here we show an example of the model application on real data from the Pan-Cancer Analysis of -->
<!--  Whole Genomes (PCAWG) cohort, specifically from a Renal Cell Carcinoma (Kidney-RCC) tumor type sample.  -->
<!-- ```{r setup, warning=FALSE} -->
<!-- library(tickTack) -->
<!-- library(gridExtra) # add in plotting function -->
<!-- library(ggplot2) # add in plotting function -->
<!-- library(bayesplot) # add in plotting function -->
<!-- library(tibble)  # check if needed -->
<!-- library(dplyr) # check if needed -->
<!-- library(bayesplot) #for plotting -->
<!-- library(ggplot2) #for plotting -->
<!-- library(stringr) #for plotting -->
<!-- library(patchwork) #for plotting -->
<!-- ``` -->
<!-- ```{r real data Kidney, echo = T, results = 'hide', warning=FALSE} -->
<!-- # Load template data -->
<!-- tickTack::pcawg_example_2 -->
<!-- data <- pcawg_example_2 -->
<!-- tolerance <- 0.1 -->
<!-- data <- tickTack::fit_h(data, max_attempts=2, INIT=TRUE, tolerance = tolerance, n_components = 0) -->
<!-- results <- data$results_timing -->
<!-- clock_assignment <- results$draws_and_summary[[3]]$summarized_results -->
<!-- input_data <- results$data$input_data -->
<!-- draws_and_summary <- results$draws_and_summary -->
<!-- log_lik_matrix_list <- results$log_lik_matrix_list -->
<!-- elbo_iterations <- results$elbo_iterations -->
<!-- results_model_selection <- model_selection_h(input_data, draws_and_summary, log_lik_matrix_list, elbo_iterations, n_components = 0) -->
<!-- best_K <- results_model_selection$best_K -->
<!-- model_selection_tibble <- results_model_selection$model_selection_tibble -->
<!-- entropy <- results_model_selection$entropy_list -->
<!-- p <- tickTack::plot_timing_h(results, best_K) -->
<!-- p -->
<!-- ``` -->
<!-- ```{r visualize, echo = T, warning=FALSE, fig.align='center'} -->
<!-- # Visualize distributions of draws from the approximate posterior -->
<!-- K = nrow(results_model_selection$model_selection_tibble) -->
<!-- S = nrow(results$data$accepted_cna) -->
<!-- ``` -->
<!-- ```{r tau, echo = T, fig.align='center'} -->
<!-- # from bayesplot -->
<!-- draws <- results$draws_and_summary[[best_K]]$draws -->
<!-- names <- paste("tau[", 1:best_K, "]", sep = "") -->
<!-- areas_tau <- mcmc_areas( -->
<!-- draws, -->
<!-- pars = names, -->
<!-- prob = 0.8, # 80% intervals -->
<!-- prob_outer = 0.95, # 99% -->
<!-- point_est = "median" -->
<!-- )+ -->
<!-- labs( -->
<!-- title = "Approximate Posterior distributions", -->
<!-- subtitle = "With median and 80% and 95% intervals" -->
<!-- )+ -->
<!-- xlim(0, 1) # + scale_x_continuous(breaks = c(1:5), labels = c("A", "B", "C", "D", "E")) -->
<!-- areas_tau -->
<!-- ``` -->
<!-- ```{r weights, echo = T, fig.height = 6, fig.width = 12, fig.align = "center"} -->
<!-- # from bayesplot -->
<!-- intervals_weigths_per_tau <- list() -->
<!-- for (k in 1:best_K){ -->
<!--   names_weights <- paste("w[",1:S,",", k, "]", sep = "") -->
<!--   p <- mcmc_intervals(draws, pars = names_weights, point_est = "median", prob = 0.8, prob_outer = 0.95)+ -->
<!--     labs( -->
<!--     title =  str_wrap( paste0("Posterior distributions of the weigths for tau ",k), width = 30 + K + sqrt(S)), -->
<!--     subtitle = "With median and 80% and 95% intervals" -->
<!--     ) -->
<!--    intervals_weigths_per_tau[[k]] <- ggplotGrob(p) -->
<!-- } -->
<!-- p <- gridExtra::grid.arrange(grobs = intervals_weigths_per_tau, ncol= best_K) #add global title -->
<!-- ``` -->
<!-- ```{r source plot, include=TRUE} -->
<!-- source("./plotting_function.R", local = knitr::knit_global()) -->
<!-- ``` -->
<!-- ```{r elbo, echo = T, fig.height = 20, fig.width = 12, fig.align = "center"} -->
<!-- # visualize elbo -->
<!-- p_elbo <- list() -->
<!-- for (i in 1:K){ -->
<!--   p_elbo[[i]] <- plot_elbo(elbo_iterations, i) + ggtitle(paste0("K = ", i)) -->
<!--   saveRDS(p_elbo, paste0("./plots/elbo_vs_iterations_",i,".rds")) -->
<!-- } -->
<!-- p_elbo <- gridExtra::grid.arrange(grobs = p_elbo, nrow=K)  #add global title -->
<!-- p_elbo -->
<!-- ``` -->
<!-- ```{r model, echo = T, fig.height = 8, fig.width = 12, fig.align = "center"} -->
<!-- # visualize model selection scores -->
<!-- p_model_selection <- plotting_model_selection(model_selection_tibble, best_K, entropy) -->
<!-- p_model_selection -->
<!-- cat("Best K =",best_K) -->
<!-- ``` -->
<!-- ```{r plots, echo = T, fig.height = 18, fig.width = 12, fig.align = "center"} -->
<!-- plot_model_selection_inference <- list() -->
<!-- for (i in 1:K){ -->
<!--   plot_model_selection_inference[[i]] <- tickTack::plot_inference_h(results, i) + ggtitle(paste0("K = ", i)) -->
<!-- } -->
<!-- plot_model_selection_inference <- gridExtra::grid.arrange(grobs = plot_model_selection_inference, nrow=K) #add global title -->
<!-- plot_model_selection_inference -->
<!-- ``` -->
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
